{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"toc_visible":true,"authorship_tag":"ABX9TyPuSXFTfwR6WpimcJUlUjtd"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["# **Chapter 7: Transcriptional Regulation**"],"metadata":{"id":"wdnk7q-xjXZI"}},{"cell_type":"markdown","source":["## ***Learning Objectives:***\n","\n","* Identify the major motifs in gene transcriptional regulatory networks\n","* Calculate the dynamic properties associated with these motifs using multiple\n","methods\n","* Compare model outputs with corresponding experimental data\n","* Understand how simpler motifs combine to form larger, more complex motifs"],"metadata":{"id":"jgo7ZJg_jiHE"}},{"cell_type":"markdown","source":["You’ve seen most of the commonly used approaches to modeling biological networks,\n","and you’ve applied these approaches to some relatively simple circuits. Now it’s time to\n","extend these techniques to larger, more complicated biological systems. Section II will\n","lead you through the strategies that researchers have used to tackle three kinds of\n","biological networks at a larger scale: transcriptional regulation, or the modification of\n","transcription factor activity to affect gene expression; **signal transduction**, or how cells\n","\n","sense their surrounding environments and initiate appropriate responses; and carbon-\n","energy **metabolism**, which breaks down nutrients from the environment to produce all of\n","\n","the building blocks to make a new cell.\n","To address these topics, I’ll have to cover much more biology. We’ll also learn another\n","analysis method or two along the way, but by and large we will be applying methods that\n","you have already learned to more difficult problems. You will soon find that you are\n","already well equipped to understand and critique existing models as well as to create your\n","own!"],"metadata":{"id":"OwJf4gJNjr_Y"}},{"cell_type":"markdown","source":["## **Section 7.1. Transcriptional regulation and complexity**"],"metadata":{"id":"36xYYbEWcc47"}},{"cell_type":"markdown","source":["As an example, let’s consider gene expression again. We’ve already looked at regulation\n","extensively in Section I, but this chapter is going to add a new layer of complexity,\n","including the interactions of multiple transcription factors to produce more complex\n","expression dynamics."],"metadata":{"id":"Bef3QvRScelh"}},{"cell_type":"markdown","source":["Let’s start with one of the most exciting events in the history of biology – the publication\n","of the human genome in 2001. David Baltimore, a preeminent biologist and Nobel\n","laureate, commented on the event with this quote: “I’ve seen a lot of exciting biology\n","emerge over the past 40 years. But chills still ran down my spine when I first read the\n","paper that describes the outline of our genome ...” (see Baltimore in Recommended\n","Reading; reprinted by permission from Macmillan Publishers Ltd: *Nature*, 2001)."],"metadata":{"id":"ZAmDswqfcm-x"}},{"cell_type":"markdown","source":["At the time, I remember that a number of interesting aspects of the sequence had us\n","talking. As Baltimore wrote, “What interested me most about the genome? The number\n","of genes is high on the list ... it is clear that we do not gain our undoubted complexity\n","over worms and plants by using many more genes. Understanding what does give us our complexity... remains a challenge for the future ...” (see Baltimore in Recommended\n","Reading; reprinted by permission from Macmillan Publishers Ltd: *Nature*, 2001)."],"metadata":{"id":"FSpi8HXjcrYC"}},{"cell_type":"markdown","source":["It’s not only the number of the genes in these genomes that was surprising; many genes\n","encode proteins that are essentially the same, and carry out the same functions. Many\n","scientists had previously assumed that the differences between species depended mostly\n","on different genes: a human had human genes, a mouse had mouse genes, a fish had fish\n","genes, and a sea urchin had sea urchin genes. However, the genomic sequences of all of\n","these organisms suggested that differences in the gene complement played a much\n","smaller role than first anticipated. For example, we share nearly all of our genes with\n","mice."],"metadata":{"id":"_uC7Sd4Rcwrl"}},{"cell_type":"markdown","source":["So what makes us different? The key is not primarily in the genes themselves, but how\n","they are expressed. Humans have approximately eight times as much DNA sequence as\n","the puffer fish, but essentially the same number of genes. That “extra” DNA used to be\n","called “junk” (honestly!), but it’s now clear that most of it is functional, and its primary\n","function appears to be the regulation of gene expression."],"metadata":{"id":"PgFcjPgccybm"}},{"cell_type":"markdown","source":["Gene expression can be regulated at several points, including transcription of DNA to\n","RNA; RNA processing, localization, and degradation; translation of the mRNA transcript\n","into a peptide chain; and activity of the final protein. In this chapter, we’ll concentrate on\n","transcriptional regulation (protein activity will appear in the next chapters). As you’ll\n","remember from Figure 1.2, we introduced this control with the example of a transcription\n","factor that is bound by a small molecule, changing the protein’s affinity for its binding\n","site on the DNA. The binding of the transcription factor in turn affects the recruitment of\n","the RNA polymerase complex and, subsequently, the expression of mRNA from the\n","gene."],"metadata":{"id":"a6x6w3Sxc5JA"}},{"cell_type":"markdown","source":["![Figure 7.1](https://drive.google.com/uc?export=view&id=19-CCvFUodSrB3qmMosTz1OTcquvrR8Ux)"],"metadata":{"id":"NLOm43S8c7mj"}},{"cell_type":"markdown","source":["> **Figure 7.1. The E. coli transcription factors CRP and LacI interact on the\n","lac promoter to control gene expression.** (A) Schematic of the various inputs\n","to the lac promoter, the combinations of which determine the transcriptional\n","state of the lac operon. (B) The same system represented as a circuit diagram.\n","Pointed arrowheads denote positive regulation, while blunt arrowheads indicate\n","negative regulation."],"metadata":{"id":"mjGRpoBEc-x7"}},{"cell_type":"markdown","source":["## **Section 7.2. More complex transcriptional circuits**"],"metadata":{"id":"lKWWBa6EdCXF"}},{"cell_type":"markdown","source":["Section 7.1 contained the simplest example we could have considered; now, let’s move\n","toward more complex modeling of transcriptional regulation by considering a pair of\n","transcription factors acting on the same gene promoter."],"metadata":{"id":"eTmPu9xOdFod"}},{"cell_type":"markdown","source":["The classic real-world example of such regulation is the regulation of the lac genes,\n","whose gene products enable *E. coli* to grow on lactose, and whose expression depends on\n","two transcription factors (Figure 7.1A). E. coli prefers to eat glucose, and it won’t\n","metabolize anything else until the glucose is gone. In *E. coli*, this metabolic switching is accomplished with the transcription factor CRP (you encountered CRP in Figure 1.1).\n","CRP binds the promoters of hundreds of genes once it is bound to cAMP, a small\n","molecule whose presence indicates that none of *E. coli’s* favorite sugar sources are\n","available. The CRP-cAMP complex can then bind operator sites that control the\n","expression of genes that enable the uptake and metabolism of other carbon sources."],"metadata":{"id":"FjYQJqSSdHEb"}},{"cell_type":"markdown","source":["One of these carbon sources is lactose, a sugar characterized by a particular link between\n","galactose and glucose. Utilization of lactose depends on enzymes and a transporter, the\n","genes for which appear together on the *E. coli* chromosome as a single transcription unit\n","– an operon. The transcription of multiple genes as a single transcript is a more efficient\n","way for bacteria to coordinate gene expression. Operons are common in bacteria, but are\n","\n","largely absent from more complex organisms, which tend to rely on complex post-\n","transcriptional regulatory processes."],"metadata":{"id":"x8xsPDRSdOGZ"}},{"cell_type":"markdown","source":["The promoter of the *lac* operon contains a binding site for the CRP-cAMP complex.\n","Thus, the operon is only fully expressed in the presence of CRP-cAMP, which only\n","appears in the absence of glucose. However, what if there is no lactose in the\n","environment either? It doesn’t make sense to express the *lac* genes unless both glucose is\n","absent and lactose is present. *E. coli* addresses this problem with another transcription\n","factor: lacI (the “I” stands for “inhibitor”). Free lacI inhibits transcription by binding its\n","own operator in the lac operon promoter. However, when lactose is present in the\n","external environment, one of its metabolic products (allolactose) binds lacI, reducing\n","lacI’s binding affinity to the operator and enabling the transcription of the *lac* operon."],"metadata":{"id":"0ReG5DJfdRi-"}},{"cell_type":"markdown","source":["So now you know that there are two interactions between transcription factors and\n","operator sites upstream of the *lac* operon (Figure 7.1A). However, this information is not\n","sufficient to predict the transcriptional state of the *lac* operon, because you also have to know how these two sites interact. In our case, CRP-cAMP has to be bound to the DNA,\n","and LacI must be absent from the DNA for maximum expression (an AND relationship).\n","In other cases where two transcription factors regulate expression of the same\n","transcription unit, either one of the transcription factors may be sufficient to induce\n","expression – an OR relationship."],"metadata":{"id":"tCFCVAxWdkVH"}},{"cell_type":"markdown","source":["The biology that we’ve described in detail here is sometimes represented more compactly\n","(Figure 7.1B). Signals are shown as activating (arrow) or inhibiting (blunt arrow) the\n","activity of transcription factors, which then interact at the chromosome to determine the\n","expression of target genes. Based on this diagram, we can use a simple function (CRP\n","AND NOT LacI) to represent the interaction, but the representation of regulatory control\n","doesn’t have to be Boolean."],"metadata":{"id":"Ucr4nn86dsPS"}},{"cell_type":"markdown","source":["As you can imagine, the regulation of transcription can be much more complicated in\n","mammalian systems, with some promoters more than 10 kb long (~100x longer than the\n","typical bacterial promoter), and significantly longer than the gene itself. There are also\n","many more factors that interact at the promoter, ~30 on average in some systems. We’ll\n","consider an example in Chapter 8."],"metadata":{"id":"z-KAG9TzdoTw"}},{"cell_type":"markdown","source":["## **Section 7.3. The transcriptional regulatory feed-forward motif**"],"metadata":{"id":"OUN0pkPKdvXD"}},{"cell_type":"markdown","source":["To move toward such complex networks, let’s discuss more advanced regulation\n","architectures in E. coli. Earlier, we examined a representation of the E. coli\n","transcriptional regulatory network (Figure 1.1), and we spent a considerable amount of\n","time studying the most common motif of that network: simple autoregulation, where the\n","protein product of a given gene regulates the expression of that gene. This motif was the\n","only significant motif that was identified in the regulatory network when only a single\n","gene was considered; when two genes were considered, no additional motifs of\n","significance were found (see Rosenfeld et al. in Recommended Reading).\n"],"metadata":{"id":"ZoGAwlLAdxx_"}},{"cell_type":"markdown","source":["Consideration of three-gene combinations, however, revealed a very interesting motif\n","(Figure 7.2). Here, transcription factor X regulates the expression of two other proteins,\n","Y and Z. However, Y is also a transcription factor, and X and Y both regulate the\n","expression of Z. This motif was called a **feed-forward loop** (control theorists: remain\n","calm. This utilization of “feed-forward” does not correspond well to the usage in control\n","theory. That’s okay.)."],"metadata":{"id":"gsNK7nnld2eA"}},{"cell_type":"markdown","source":["![Figure 7.2](https://drive.google.com/uc?export=view&id=1qTKJCAM4RimwW3Xvt0Qmd5tJRdQIuhKc)"],"metadata":{"id":"Ke27yfaKd6P7"}},{"cell_type":"markdown","source":["> **Figure 7.2. The feed-forward loop.** This motif was found much more\n","frequently in biological transcription networks than in randomly generated\n","networks. The transcription factor X regulates the expression of two proteins, Y\n","and Z. However, Y also regulates Z."],"metadata":{"id":"phoQSFazd8UN"}},{"cell_type":"markdown","source":["This motif occurred far more often in the E. coli regulatory network than would be\n","expected by random chance, as was demonstrated by performing the same analysis on\n","randomized networks that had the same nodes and the same number of connections as the\n","E. coli network, but the connections were randomly scrambled. In the randomized\n","networks, an average of only about two feed-forward motifs were identified, but in the\n","real E. coli network, there were 42 (Douglas Adams, are you reading this?!?)."],"metadata":{"id":"ps9ybjO3d_-E"}},{"cell_type":"markdown","source":["So what makes feed-forward loops special? The short answer is that they exhibit\n","fascinating and useful dynamics in terms of how they express their genes. To arrive at\n","this answer, we first have to recognize that there are many kinds of feed-forward loops.\n","Figure 7.3 illustrates all eight of the possible combinations of negative and positive\n","regulation. Each combination has a direct arm from X to Z (top) and an indirect arm\n","through Y (bottom), and either arm can exert a positive or negative influence on the\n","expression of Z."],"metadata":{"id":"n9M04xLIeGyA"}},{"cell_type":"markdown","source":["Now, notice that for some of the combinations, both arms exhibit the same kind of\n","influence, whether positive or negative. For other combinations, the arms have different\n","influences: one arm regulates positively and the other negatively. When the direct and\n","indirect arms regulate in the same way, the combination is called **internally consistent**\n","(or sometimes, “coherent”), meaning that the direct regulation from X to Z matches the\n","indirect regulation through Y. For example, on the upper left loop in Figure 7.3, X has a\n","positive and direct effect on Z, but it also exerts a positive and indirect effect because X\n","positively regulates Y, and Y positively regulates Z. Keep in mind that if Y has a\n","negative effect on Z, and X has a negative effect on Y, then the overall indirect effect of\n","X on Z is positive. Four of the feed-forward loop combinations are internally consistent,\n","and the other half are internally inconsistent; the direct and indirect actions of X on Z are\n","opposite."],"metadata":{"id":"WIIaSzuGeIas"}},{"cell_type":"markdown","source":["![Figure 7.3](https://drive.google.com/uc?export=view&id=18lw6Vh4Qj0puiYokn21pytrknx1rLNRR)\n"],"metadata":{"id":"p483QGg5eMQi"}},{"cell_type":"markdown","source":["> **Figure 7.3. Possible instances of regulatory feed-forward loops.** Both arms of\n","internally consistent feed-forward loops exert the same type of control, whether\n","positive or negative. Internally inconsistent loops have one positive arm and one\n","negative arm. As in Figure 7.2, pointed arrowheads represent positive regulation\n","and blunt arrowheads denote negative regulation. The red box highlights the two\n","instances that were specifically overrepresented in both *E. coli* and yeast. Alon, U.\n","*An Introduction to Systems Biology: Design Principles of Biological Circuits.*\n","Chapman & Hall/CRC, 2007. Reproduced with permission of TAYLOR &\n","FRANCIS GROUP LLC in the format Republish in a book via Copyright Clearance\n","Center."],"metadata":{"id":"mvpVmfh2eN36"}},{"cell_type":"markdown","source":["## **7.4. Boolean analysis of the most common internally consistent feed-forward motif identified in *E. coli***"],"metadata":{"id":"OSgh_Qy9eYdx"}},{"cell_type":"markdown","source":["Going back to the *E. coli* network, only two of the feed-forward loop submotifs, one\n","coherent and one incoherent, were actually found to significantly contribute to the\n","regulatory network (red box in Figure 7.3). First let’s consider the coherent feed-forward\n","loop (Figure 7.4). Here, a gene $g_x$ is transcribed to produce a protein $p_x$, which can then\n","be activated by a signal $s_x$. The active protein ${p_x}^*$ can bind the promoters for genes gy\n","and gz. The product of gy is $p_y$, which can be activated by sy to produce ${p_y}^*$. The\n","combination of ${p_x}^*$ and ${p_y}^*$ at the gz promoter leads to the production of $p_z$."],"metadata":{"id":"EKiDTarWeese"}},{"cell_type":"markdown","source":["![Figure 7.3](https://drive.google.com/uc?export=view&id=1gtNEqyMdPTYPE62gxwsK4fnVE8Wu9EqE)"],"metadata":{"id":"GzLJFvPRe4vZ"}},{"cell_type":"markdown","source":["> **Figure 7.4. Gene regulatory circuit diagram for the most common\n","internally consistent feed-forward loop in *E. coli.*** Notation is as in Figure\n","7.1B, but notice the arrow (red), which highlights the positive regulation of\n","py by px."],"metadata":{"id":"vaFc1Rc2e6sY"}},{"cell_type":"markdown","source":["Let’s look at the dynamics of this circuit. A good place to start is with a Boolean\n","analysis, similar to Chapter 2. Our inputs will be $s_x$ and $s_y$, and we’ll assume that all of\n","the genes are present $(g_x = g_y = g_z = 1)$. Furthermore, as in earlier chapters, let’s assume\n","that sufficient activation of $p_x$ and $p_y$ occurs essentially instantaneously if $s_x$ and $s_y$ are\n","present. Our equations are then reduced to:"],"metadata":{"id":"2sQQr94pfAa3"}},{"cell_type":"markdown","source":["$\\text{Activation}_{px}$ = IF $s_x$\n","\n","${p_x}^*$ = IF $\\text{Activation}_{px}$ AFTER SOME TIME\n","\n","$\\text{Activation}{py}$ = IF ${p_x}^*$ AND $s_y$\n","\n","${p_y}^*$ = IF $\\text{Activation}_{py} AFTER SOME TIME\n","\n","$\\text{Expression}_{pz}$ = IF ${p_x}^*$ AND ${p_y}^*$\n","\n","$p_z$ = IF $\\text{Expression}_{pz}$ AFTER SOME TIME"],"metadata":{"id":"2-ggIqGUfNkv"}},{"cell_type":"markdown","source":["Using these equations, we can draw the (partial) state diagram shown in Figure 7.5. We\n","begin with initial conditions of no active protein ${p_x}^*$, ${p_y}^*$, or expressed protein pz, and\n","sudden addition of signals $s_x$ and $s_y$. Addition of $s_x$ leads to the activation of px to ${p_x}^*$,\n","followed by the expression and activation of ${p_y}^*$ (expression is the step that takes more\n","time; both processes are lumped into our equations). Finally,once ${p_x}^*$ and ${p_y}^*$ are active,\n","$p_z$ is expressed to yield the steady state of the circuit."],"metadata":{"id":"iv5aL1wcgFyr"}},{"cell_type":"markdown","source":["![Figure 7.5](https://drive.google.com/uc?export=view&id=1FT2tpH9naHhuVRprzneoknorI7TVqzeg)"],"metadata":{"id":"_uwY223Hghpv"}},{"cell_type":"markdown","source":["> **Figure 7.5. State matrix (A) and dynamics (B) of the coherent feed-forward\n","loop in Figure 7.4, in the presence of both signals.** After $p_x$ is activated to ${p_x}^*$,\n","$p_y$ can be expressed and subsequently activated, after which pz can be expressed.\n","Note that the increase in ${p_x}^*$ occurs after stimulus is added because the Boolean\n","rule is AFTER SOME TIME."],"metadata":{"id":"_Jtta2jNgjbR"}},{"cell_type":"markdown","source":["The dynamics of this circuit appear in Figure 7.5B. Notice that there are two periods of\n","expression between activation of the circuit and the expression of $p_z$. If the circuit were a\n","simple induction circuit with no feedback (if only ${p_x}^*$ regulated the expression of $p_z$),\n","\n","then $p_z$ would be expressed in roughly one-half the time that it takes with the feed-\n","forward motif. The motif therefore increases the time required for expression of the\n","\n","target gene."],"metadata":{"id":"BR5OrnyBgzEi"}},{"cell_type":"markdown","source":["Figure 7.5 is our analysis of what happens when the circuit is initially inactive (no\n","stimuli) and we then activate it; now, let’s contrast that behavior with a circuit that is\n","initially active and is then inactivated. Let’s begin with the state-matrix approach shown\n","in Figure 7.6A. When the stimulus is removed, ${p_x}^*$ becomes rapidly deactivated to $p_x$.\n","The expression of both $p_y$ and $p_z$ depends on ${p_x}^*$, and so neither protein can be produced.\n","As a result, both proteins decay over the same time period (Figure 7.6B)."],"metadata":{"id":"izLxKFvRg9zi"}},{"cell_type":"markdown","source":["Thus, when this system is activated, there are two time steps: one for the expression and\n","activation of ${p_y}^*$ and one for the expression of $p_z$. However, when the system is\n","deactivated, there is only a single time step, because ${p_y}^*$ and $p_z$ are removed\n","simultaneously. This incoherent feed-forward loop created a switch with different ON\n","and OFF times! Why is this strategy useful? Uri Alon, whose team at the Weizmann\n","Institute of Science originally identified these motifs, gives the example of the door of an\n","elevator. It’s very important that the door be safe, which means that it should start to\n","close slowly but stop closing quickly, for example at the instant that someone’s foot\n","triggers the safety mechanism."],"metadata":{"id":"ZUpDXQWAhI3o"}},{"cell_type":"markdown","source":["![Figure 7.6](https://drive.google.com/uc?export=view&id=11EzQsMBNRpWK0drfFiuQyTIRjH7un9HS)"],"metadata":{"id":"pN2ZwQFxhWNB"}},{"cell_type":"markdown","source":["> **Figure 7.6. State matrix (A) and the dynamics (B) illustrate the de-\n","activation of the feed-forward loop in Figure 7.4.** In this case, $s_x$ is switched\n","to zero, which leads to deactivation of ${p_x}^*$ (red arrow 1), followed by a halt in\n","the expression of $p_y$ and $p_z$. Compare these figures with Figure 7.5."],"metadata":{"id":"zHSp3cUGhX-z"}},{"cell_type":"markdown","source":["## **Section 7.5. An ODE-based approach to analyzing the coherent feed-forward loop**"],"metadata":{"id":"j829HSxshkcK"}},{"cell_type":"markdown","source":["We’ve discussed how to analyze this circuit using a Boolean approach; a parallel\n","approach (favored by Uri Alon, see Recommended Reading) would be to write ODEs as\n","follows. We’ll begin with $d[p_y]/dt$ in the usual formulation:"],"metadata":{"id":"qOC8X-49hnSg"}},{"cell_type":"markdown","source":["> <h3> $\\frac{d[p_y]}{dt} = prod - loss$\n","\n","*(Equation 7.1)*"],"metadata":{"id":"yrcH21gShsNm"}},{"cell_type":"markdown","source":["We still represent the loss term as proportional to the amount of $d[p_y]/dt$. For production,\n","we will invoke a **threshold concentration**: transcription of $p_y$ can only begin once $[{p_x}^*]$\n","reaches a certain value. We will indicate threshold notations with the notation $K_{ab}$ where\n","a denotes the transcription factor and b the target gene; therefore, we will add a $K_{xy}$ term\n","to Equation 7.1. We further assume that the expression of $p_y$ is maximal when $[{p_x}^*]$ is\n","greater than the threshold constant $K_{xy}$, and that otherwise the expression of $p_y$ is equal to\n","zero. Adding these details to our equation, we obtain:"],"metadata":{"id":"BNrT3_N-h3Bw"}},{"cell_type":"markdown","source":["> <h3> $\\frac{d[p_y]}{dt} = k_{yprod}Θ ([{p_x}^*] > K_{xy}) - k_{ydeg}[p_y]$\n","\n","*(Equation 7.2)*"],"metadata":{"id":"2-vM7fUxiQNv"}},{"cell_type":"markdown","source":["where the function θ (statement) is equal to one if the statement in parentheses is true,\n","and equal to zero if the statement is false. For the case in which $s_x$ has been added at a sufficient concentration for activation, θ = 1, and Equation 7.2 is reduced to:"],"metadata":{"id":"-BWagCkWivWv"}},{"cell_type":"markdown","source":["> <h3> $\\frac{d[p_y]}{dt} = k_{yprod} - k_{ydeg}[p_y]$\n","\n","*(Equation 7.3)*"],"metadata":{"id":"JcDTRdsNi1Zs"}},{"cell_type":"markdown","source":["We’ve solved equations like this before (in Chapter 3, for example), and so you should be\n","able to show that the solution of this equation is:"],"metadata":{"id":"j4svmfLejBNb"}},{"cell_type":"markdown","source":["> <h3> $[p_y](t) = [p_{y,ss}](1 - e^{-k_{ydeg}t})$\n","\n","*(Equation 7.4)*"],"metadata":{"id":"czu4Cne6jDGC"}},{"cell_type":"markdown","source":["where $[p_{y,ss}]$ is determined by:\n","\n","> <h3> $0 = k_{yprod} - k_{ydeg}[p_{y,ss}]$\n","\n","*(Equation 7.5)*"],"metadata":{"id":"mJ0Pg3tJjUGj"}},{"cell_type":"markdown","source":["and therefore:\n","\n","> <h3> $ [p_{y,ss}] = \\frac{k_{yprod}}{k_{ydeg}}$\n","\n","*(Equation 7.6)*"],"metadata":{"id":"P_kgIZ8rjkJI"}},{"cell_type":"markdown","source":["In general, the reactions that lead to a transcription factor being activated happen at a\n","significantly faster rate (on the order of seconds or less) than the rate of gene expression\n","(minutes). As a result, for our purposes here we’ll assume that the transition from py to\n","${p_y}^*$ is very fast in the presence of sy, and so in this case, $p_y(t) = [p_y]^*(t)$."],"metadata":{"id":"qW7F8UUkjx7r"}},{"cell_type":"markdown","source":["The equation for $p_z$ is similar to the equation for $p_y$, but in this case the production of $p_z$ is\n","based on two conditions occurring simultaneously:"],"metadata":{"id":"Q5_RmvBQj6aH"}},{"cell_type":"markdown","source":["> <h3> $ \\frac{d[p_z]}{dt} = k_{zprod}Θ([{p_x}^*] > K_{xz})Θ([{p_y}^*] > K_{yz}) - k_{zdeg}[p_z]$\n","\n","*(Equation 7.7)*"],"metadata":{"id":"UExOo6VUkA9j"}},{"cell_type":"markdown","source":["When ${p_x}^*$and ${p_y}^*$ are sufficiently large, the solution is similar to that for [$p_y$]:"],"metadata":{"id":"aM0LsGPtkkrU"}},{"cell_type":"markdown","source":["> <h3> $[p_z](t) = [p_{z,ss}](1 - e^{-k_{zdeg}t}), [p_{z,ss}] = \\frac{k_{zprod}}{k_{zdeg}}$\n","\n","*(Equation 7.8)*"],"metadata":{"id":"yeUw-GDhkydF"}},{"cell_type":"markdown","source":["The response of the system to a sudden addition of $s_x$ and $s_y$ is depicted in Figure 7.7.\n","After $s_x$ is added, the expression of py and consequently ${p_y}^*$ increases until [${p_y}^*$] reaches the threshold $K_{yz}$. At this time, $p_z$ expression is induced."],"metadata":{"id":"EWnbtd6-lYc3"}},{"cell_type":"markdown","source":["![Figure 7.7](https://drive.google.com/uc?export=view&id=1MEKG_JvFERUl6rE2DimTqWMDOkxDgmga)"],"metadata":{"id":"hNVFaQP_llUi"}},{"cell_type":"markdown","source":["> **Figure 7.7. Dynamic response of the coherent feed-forward loop in\n","Figure 7.4 to (A) sudden addition or (B) sudden removal of $s_x$ at time\n","zero.** Notice that there is a delay (Ton) between expression changes in ${p_y}^*$\n","and $p_z$ when $s_x$ is added, but not when $s_x$ is removed, as we also observed\n","in our Boolean analysis (Figures 7.5 and 7.6). Remember that $p_y$ = ${p_y}^*$\n","due to an assumption of rapid $p_y$ activation. Modified from Alon, U. *An\n","Introduction to Systems Biology: Design Principles of Biological Circuits.*\n","Chapman & Hall/CRC, 2007. Reproduced with permission of TAYLOR\n","& FRANCIS GROUP LLC in the format Republish in a book via\n","Copyright Clearance Center."],"metadata":{"id":"gxwgrnWalnf-"}},{"cell_type":"markdown","source":["Let’s consider the time required for $[{p_y}^*]$ to reach the threshold $K_{yz}$, as this delay was found to be the most notable aspect of this circuit’s dynamic response. As shown in Figure 7.8, we’ll call this delay Ton, and determine it from Equation 7.4 (remembering\n","that $[p_y](t) = [p_y]^*(t)$ if $s_y$ is present):"],"metadata":{"id":"ej4c0iJjl-Oy"}},{"cell_type":"markdown","source":["> <h3> $[{p_y}^*](T_{on}) = [p_{y,ss}](1 - e^{-k_{ydeg}T_{on}})$\n","\n","*(Equation 7.9)*"],"metadata":{"id":"Pbx4W_aBmL1M"}},{"cell_type":"markdown","source":["Since $[{p_y}^*](T_{on}) = K_{yz}$, we can solve for $T_{on}$:\n","\n","> <h3> $K_{yz} = [p_{y,ss}](1 - e^{-k_{ydeg}T_{on}})$\n","\n","*(Equation 7.10)*\n","\n"],"metadata":{"id":"kHv8PxSSmdlC"}},{"cell_type":"markdown","source":["> <h3> $T_{on} = \\frac{1}{k_{ydeg}}ln(\\frac{1}{1 - \\frac{K_{yz}}{[p_{y,ss}]}})$\n","\n","*(Equation 7.11)*"],"metadata":{"id":"_OGZ0HFfnB7H"}},{"cell_type":"markdown","source":["From Equation 7.11, you can see that when $[p_{y,ss}]$ is much larger than $K_{yz}$, $K_{yz}/[p_{y,ss}]$\n","approaches zero, which means that $T_{on}$ will also reduce to zero for a given kydeg. As the value of $[p_{y,ss}]$ approaches the threshold value $K_{yz}$, the logarithm term of Equation 7.11 increases rapidly, meaning that the value of $T_{on}$ increases dramatically as well."],"metadata":{"id":"2jaqIqIjnZJz"}},{"cell_type":"markdown","source":["## **Section 7.6. Robustness of the coherent feed-forward loop**"],"metadata":{"id":"jzrRMzJhnpyK"}},{"cell_type":"markdown","source":["As you can see, the overall conclusions that we drew in Section 7.5 using ODEs were\n","very similar to the results of our Boolean-based approach in Section 7.4. However,\n","ODE-based formulation of the model also allows us to demonstrate another interesting\n","aspect of this coherent feed-forward circuit: its **robustness**. In this context, robustness\n","means that the system won’t change much in response to a small perturbation. As an\n","example, let’s say that our circuit is exposed only briefly to $s_x$. As before, $p_y$ begins to be\n","expressed, but it doesn’t reach the critical threshold for pz expression, and thus $p_z$ is never\n","expressed (Figure 7.8). The robustness of the system is tuned by the value of the\n","threshold: a higher threshold takes longer for the $p_y$ value to obtain, and so expression of\n","$p_z$ would be robust to even longer stimulus times."],"metadata":{"id":"ae2i7VuKnsLA"}},{"cell_type":"markdown","source":["![Figure 7.8](https://drive.google.com/uc?export=view&id=1RFNoZXQYwcR8vGa8DDvje2Tz-AGv1D1P)"],"metadata":{"id":"9Pw_gZPvn4lg"}},{"cell_type":"markdown","source":["> **Figure 7.8. Response of the feed-forward loop in Figure 7.4 to a brief pulse\n","of $s_x$.** The pulse of stimulus (highlighted by an arrow in the bottom plots) leads\n","to only brief activation of $p_x$, which in turn leads to a maximum expression of $p_y$\n","that is below the threshold $K_{yz}$ (dashed line). As a result, $p_z$ expression is never\n","induced. For contrast, a longer pulse of $s_x$ and its consequences are shown at\n","right (and in Figure 7.7A). Adapted by permission from Macmillan Publishers\n","Ltd: Shen-Orr, S. S., Milo, R., Alon, U. Network motifs in the transcriptional\n","regulation network of *Escherichia coli.* Nature Genetics. 2002. **31**(1):64-8."],"metadata":{"id":"FG9k4t2fn7Sc"}},{"cell_type":"markdown","source":["## **Section 7.7. Experimental interrogation of the coherent feed-forward loop**"],"metadata":{"id":"kKLRM7CmoKTH"}},{"cell_type":"markdown","source":["Having mathematically analyzed this coherent feed-forward circuit in detail, Alon’s team\n","decided to see whether the experimental data from a naturally occurring feed-forward\n","circuit in *E. coli* actually exhibited the dynamics that theory predicted. They focused on the *ara* genes, which are regulated by the transcription factors AraC and CRP. AraC is regulated transcriptionally by CRP, and is activated in the presence of arabinose (Figure\n","7.9A). By keeping the bacteria in an arabinose environment, and suddenly adding or\n","“removing” cAMP (see figure legend), the team replicated the situation that they\n","modeled. To enable comparison, the team simultaneously considered a control circuit\n","that responded directly to CRP but not to AraC (grey lines in Figure 7.9B and C).\n","Consistent with the theory shown in Figure 7.7, the expression of the *ara* genes was\n","delayed when cAMP was added (Figure 7.9B), but not when cAMP was “removed”\n","(Figure 7.9C) – an elegant demonstration of their theory!"],"metadata":{"id":"h7sDoC6moNo6"}},{"cell_type":"markdown","source":["![Figure 7.9](https://drive.google.com/uc?export=view&id=1u4aOJicLO1TQvv2ujP4HBvr00kvLBlVW)"],"metadata":{"id":"xa2YER1ToXYY"}},{"cell_type":"markdown","source":["> **Figure 7.9. Monitoring promoter states when turning on and off circuits with\n","(black) and without (grey) feed-forward components.** The feed-forward circuit\n","(A) and a control circuit (based on the *lac* operon) were (B) turned on by adding\n","saturating amounts of cAMP to growing cells, and (C) turned off by “removing”\n","cAMP (actually by adding saturating glucose, which inactivates CRP). The\n","promoters were transcriptionally fused to GFP, which acted as a reporter of\n","promoter activity. Note that turning on the circuit is delayed in the feed-forward\n","loop (B), but the two circuits have similar dynamics when turning off (C).\n","Mangan, S., Zaslaver, A., Alon, U. The coherent feedforward loop serves as a\n","sign-sensitive delay element in transcription networks. *Journal of Molecular\n","Biology.* 2003. **334**(2): 197-204. Reprinted with minor modifications with\n","permission from Elsevier."],"metadata":{"id":"WkdKUaRIoZPX"}},{"cell_type":"markdown","source":["## **Section 7.8. Changing the interaction from an AND to an OR relationship**"],"metadata":{"id":"PunxPb4Oog0T"}},{"cell_type":"markdown","source":["In our analysis of the coherent feed-forward circuit in Sections 7.4-7, we focused on an\n","AND relationship between the transcription factors controlling gene expression: both ${p_x}^*$ AND ${p_y}^*$ were required. Several other relationships are possible that can lead to\n","differences in network behavior. For example, let’s change our current circuit such that\n","the interaction at the promoter changes from an AND interaction to an OR interaction.\n","We can use our Boolean toolbox for a quick analysis. Our equations for $\\text{Activation}_{px}$, ${p_x}^*$, $\\text{Activation}_{py}$, ${p_y}^*$, and pz remain the same as in Section 7.4, but the equation for $p_z$\n","expression becomes:"],"metadata":{"id":"mavJUaEEoj3D"}},{"cell_type":"markdown","source":["$\\text{Expression}_{pz} = \\text{IF } {p_x}^* \\text{OR } {p_y}^*$"],"metadata":{"id":"fU2HY_9mpF_-"}},{"cell_type":"markdown","source":["![Figure 7.9](https://drive.google.com/uc?export=view&id=14RrScCJRrL72qHohpW795u8-ApI5YWpj)"],"metadata":{"id":"N0QRs-yYqCTd"}},{"cell_type":"markdown","source":["> **Figure 7.10. The state matrix (A) and dynamics (B) for the OR circuit.**\n","Note the delay when the stimulus sx is removed, but not when it is added."],"metadata":{"id":"jzZpHAkiqEK-"}},{"cell_type":"markdown","source":["The state matrix and dynamics for the OR circuit appear in Figure 7.10. Notice that in\n","this case, the delay occurs when the stimulus is removed! This simple change in the\n","interaction at the promoter therefore determines whether the delay occurs in the\n","expression of $p_z$ or in the decay of $p_z$."],"metadata":{"id":"BwXUsepmqILE"}},{"cell_type":"markdown","source":["Once again, Alon’s team set out to experimentally verify these predictions. They focused\n","their investigation on some of the genes that regulate expression of the bacterial\n","flagellum; these genes are naturally regulated by a coherent feed-forward circuit with an\n","OR interaction. As shown in Figure 7.11, their experimental results strongly agreed with\n","the theory we worked through above."],"metadata":{"id":"t9PsraEKqMU8"}},{"cell_type":"markdown","source":["![Figure 7.11](https://drive.google.com/uc?export=view&id=10-p4ZZl4UrW2v2l8w5h8cbyuuOjPGbWA)"],"metadata":{"id":"D26klcqOqNAE"}},{"cell_type":"markdown","source":["> **Figure 7.11. Experimental validation of feed-forward dynamics for the coherent\n","feed-forward loop with an OR interaction at the promoter. (A) The *E. coli*\n","flagellar system, a control circuit with two inputs and an OR interaction. Both of\n","these circuit types occur naturally in *E. coli*. (B) For experimental validation, the\n","production of FlhDC is controlled with a promoter that is induced by the addition of\n","arabinose (not the native promoter), which serves as $s_x$. The signal $s_y$ comes from a\n","checkpoint system that monitors the production of a component of the flagellum. The\n","promoter controlling the *fli* sequences at the bottom of the circuit is fused to GFP as a\n","reporter. Here, the “on” step is similar to that for a feed-forward circuit, which occurs\n","when FliA is deleted. (C) Turning the circuit off (by shifting the cells into medium\n","without arabinose) is delayed for the feed-forward circuit when FliA is deleted (note\n","the similarity to turning off the circuit in Figure 7.9). Adapted with permission from\n","Macmillan Publishers Ltd: Kalir, S., Mangan, S., Alon, U. A coherent feed-forward\n","loop with a SUM input function prolongs flagella expression in *Escherichia coli.\n","Molecular Systems Biology.* Epub 2005 Mar 29. doi: 10.1038/msb4100010."],"metadata":{"id":"SeXMvN1iqPUa"}},{"cell_type":"markdown","source":["## <u> **Practice Problem 7.1** </u>"],"metadata":{"id":"4hMkTBtHqghS"}},{"cell_type":"markdown","source":["*Now that we’ve analyzed a coherent feed-forward loop from E. coli, let’s consider the\n","primary incoherent feed-forward loop in Figure 7.12. Draw a state diagram and\n","calculate the dynamic response for the case in which $s_x$ and $s_y$ are suddenly added.*"],"metadata":{"id":"ORMZjeRuqmFA"}},{"cell_type":"markdown","source":["![Figure 7.9](https://drive.google.com/uc?export=view&id=10F7ApyC-4ERbSG-iTtce5ujGLIDS43hK)"],"metadata":{"id":"7dzWduEVqqkC"}},{"cell_type":"markdown","source":["> **Figure 7.12. An incoherent feed-forward loop, where px exerts a\n","positive influence on $p_y$ and $p_z$, but $p_y$ has a negative influence on $p_z$.**"],"metadata":{"id":"4qhcKNMUqsYb"}},{"cell_type":"markdown","source":["**Solution:** The regulatory rules for this feed-forward loop can be written as:"],"metadata":{"id":"WsI35i7Rq1B-"}},{"cell_type":"markdown","source":["\n","$\n","\\text{Activation}_{px} = \\text{IF } s_x \\\\\n","p_x^* = \\text{IF Activation}_{px} \\text{ AFTER SOME TIME} \\\\\n","\\text{Activation}_{py} = \\text{IF } p_x^* \\text{ AND } s_y \\\\\n","p_y^* = \\text{IF Activation}_{py} \\text{ AFTER SOME TIME} \\\\\n","\\text{Expression}_{pz} = \\text{IF } p_x^* \\text{ AND NOT } p_y^* \\\\\n","p_z = \\text{IF Expression}_{pz} \\text{ AFTER SOME TIME}\n","$\n"],"metadata":{"id":"wlhhoZkXpaQj"}},{"cell_type":"markdown","source":["Using these rules, we draw a state matrix (Figure 7.13A) and plot the resulting dynamics\n","(Figure 7.13B)."],"metadata":{"id":"xdVA_9KNq3qi"}},{"cell_type":"markdown","source":["![Figure 7.13](https://drive.google.com/uc?export=view&id=1EMxjjQIpUr8_QVGFVEBUu11ujgXBnMxN)"],"metadata":{"id":"obUoFSbHq5A_"}},{"cell_type":"markdown","source":["> **Figure 7.13. The state matrix (A) and dynamics (B) for the incoherent feed-\n","forward loop in Figure 7.12.** Notice that the expression of pz rises, then falls\n","again, in a pulse."],"metadata":{"id":"XOejhPfuq6qN"}},{"cell_type":"markdown","source":["This circuit creates a “pulse” of $p_z$ expression! You can imagine that this response is quite useful to the cell. When real instances of these feed-forward loops were examined in $E. coli$, a circuit was identified (Figure 7.14A) in which $[p_z]$ decreases to a new, non- zero steady-state level predicted by the ODEs (Figure 7.14B); experimental data confirmed this prediction (Figure 7.14C)."],"metadata":{"id":"pvKpkofgrAHK"}},{"cell_type":"markdown","source":["![Figure 7.14](https://drive.google.com/uc?export=view&id=1IKyJ1T6DNflk68ut-50o9y4dJIs1c3Ld)"],"metadata":{"id":"QrAKqfTDrKQZ"}},{"cell_type":"markdown","source":["> **Figure 7.14. Dynamics of the incoherent feed-forward loop (A) as determined\n","using ODEs (B) and experimentally (C).** Here, the *galE* promoter is fused to GFP\n","as a reporter. $s_x$ is cAMP, which activates CRP, and sy is galactose, which causes\n","GalS to unbind from the galE promoter. Mangan, S., Itzkovitz, S., Zaslaver, A.,\n","Alon, U. The incoherent feed-forward loop accelerates the response-time of the *gal*\n","system of *Escherichia coli. Journal of Molecular Biology.* 2006. **356**, 1073-1081.\n","Reprinted with minor modifications with permission from Elsevier."],"metadata":{"id":"OwVu4rR9rL1r"}},{"cell_type":"markdown","source":["## **Section 7.9. The single-input module**"],"metadata":{"id":"jYqG2hIArYtH"}},{"cell_type":"markdown","source":["The feed-forward loop was the only motif that Alon’s team found in the all of the\n","possible three-node interaction sets (Figure 7.3). Now let’s talk about some of the other\n","motifs that were found when considering greater numbers of nodes, which are easy to\n","describe but exhibit behaviors that are a bit more complex."],"metadata":{"id":"uZ1AAQsorbxQ"}},{"cell_type":"markdown","source":["A very common motif, the **single-input module** has one regulator that is solely\n","responsible for the regulation of several genes, often including itself. The genes may be located in the same operon or the genes may be spread across the genome, in which case\n","the genes are said to be in the same **regulon** (meaning that they are regulated by the same transcription factor, but aren’t necessarily in physical proximity to the gene encoding that transcription factor). Single-input modules are very unlikely to occur in a random network, especially in the case of one transcription factor regulating 14 or 15 genes."],"metadata":{"id":"aF3MB--ardf4"}},{"cell_type":"markdown","source":["You can imagine that genes in the same regulon have coordinated expression, just as\n","genes in operons do. However, all of the genes in a given regulon are not necessarily\n","expressed at the same time in a single transcript, as is the general rule for genes in an\n","operon."],"metadata":{"id":"FttodAKIrl8x"}},{"cell_type":"markdown","source":["For example, consider the arginine biosynthesis single-input module in *E. coli* (Figure 7.15. These genes encode the enzymes required to synthesize the amino acid arginine\n","(we’ll discuss this metabolic process in more detail in Chapter 9), and they are only\n","expressed when arginine is not present in *E. coli’s* environment. The genes fall into\n","seven transcription units: the five individually expressed genes *argA, argD, argE, argF*, and *argI*, the operon encoding *argC, argB,* and *argH*, and the operon encoding *carA* and *carB*. All of these genes are regulated solely by the transcription factor ArgR, which is why they belong to the same single-input module. The gene encoding ArgR is also in the\n","regulon because it regulates its own expression. However, *argG* is technically not in the regulon, even though its expression is regulated by ArgR, because CRP is known to also control the expression of argG."],"metadata":{"id":"bTkz5qRPrmoV"}},{"cell_type":"markdown","source":["![Figure 7.14](https://drive.google.com/uc?export=view&id=1yKneBsN5AZsugwyw7YECRJb8bHOFehii)"],"metadata":{"id":"F_p6GM4Dr8Xb"}},{"cell_type":"markdown","source":["> **Figure 7.15. The arginine biosynthesis single-input motif and just-in-time\n","expression.** At the top left is the transcriptional regulatory motif in which the\n","argR gene product controls its own expression as well as that of several genes that\n","encode metabolic enzymes. These enzymes are arranged into the metabolic\n","pathways that are responsible for arginine biosynthesis (metabolite names appear\n","in boxes; enzymes that catalyze the conversion of one metabolite to another are\n","represented by circles and arrows). The expression of each enzyme is shown in\n","greyscale, with the lightest grey (boxed in red) indicating the inflection point at\n","which expression of the gene reached half of its maximum. Comparing these\n","inflection points can give you an idea of the order of gene expression. The Lux\n","protein, which is luminescent, was used here as the reporter of promoter activity.\n","Adapted by permission from Macmillan Publishers Ltd: Zaslaver, A., Mayo, A. E.,\n","Rosenberg, R., Bashkin, P., Sberro, H., Tsalyuk, M., Surette, M. G., Alon, U.\n","Just-in-time transcription program in metabolic pathways. *Nature Genetics.* 2004.\n","**36**(5): 486-91."],"metadata":{"id":"VYmO7scgr-xF"}},{"cell_type":"markdown","source":["Figure 7.15 illustrates the relationship between the genes in the single-input module, both in terms of transcriptional regulation and the metabolic pathway (boxes are metabolites, circles are enzymes). The figure also includes a time course detailing the expression of each gene when arginine is removed from *E. coli’s* surroundings."],"metadata":{"id":"KDMP78l1sFMB"}},{"cell_type":"markdown","source":["## **Section 7.10. Just-in-time gene expression**"],"metadata":{"id":"jXJyCxdZsJwT"}},{"cell_type":"markdown","source":["Interestingly, in Figure 7.15, the genes appear to be expressed roughly in the order\n","they’re needed, or **just in time**. For example, the enzymes encoded by *argB* and *argC* are induced just after the expression of *argA*, which catalyzes the step in the biosynthetic pathway immediately preceding them. This strategy seems to be a very efficient way to ensure that the enzymes are only made when absolutely required: in the absence of arginine. For a single-celled organism like *E. coli*, the energy and space required to produce and store extra proteins are extremely valuable, and so this efficiency, small as it seems, could yield important dividends in terms of growth rate."],"metadata":{"id":"P6lTYrmQsL1Q"}},{"cell_type":"markdown","source":["How is just-in-time expression achieved? Let’s consider a transcription factor px that\n","regulates the expression of three genes: $g_a, g_b$, and, you guessed it, $g_c$. We’ll use the same ODE-based framework as in Section 7.5, and we’ll assume as before that once $s_x$ is added, all of the $p_x $is rapidly activated to ${p_x}^*$. The equations for gene expression are:"],"metadata":{"id":"pWVp8648sW2L"}},{"cell_type":"markdown","source":["> <h3> $\\frac{d[p_x]}{dt} = k_{\\text{prod}} - k_{\\text{xdeg}} [p_x]\n","$\n","\n","*(Equation 7.12)*"],"metadata":{"id":"d4zwjA5Bsq7W"}},{"cell_type":"markdown","source":["> <h3> $\\frac{d[p_a]}{dt} = k_{\\text{aprod}} \\theta \\left( [p_x^*] > K_{xa} \\right) - k_{\\text{adeg}} [p_a]$\n","\n","*(Equation 7.13)*"],"metadata":{"id":"a3p0FxE-tC81"}},{"cell_type":"markdown","source":["> <h3> $\\frac{d[p_b]}{dt} = k_{\\text{bprod}} \\theta \\left( [p_x^*] > K_{xb} \\right) - k_{\\text{bdeg}} [p_b]$\n","\n","*(Equation 7.14)*"],"metadata":{"id":"AS16JVv1tISw"}},{"cell_type":"markdown","source":["> <h3> $\\frac{d[p_c]}{dt} = k_{\\text{cprod}} \\theta \\left( [p_x^*] > K_{xc} \\right) - k_{\\text{cdeg}} [p_c]$\n","\n","*(Equation 7.15)*"],"metadata":{"id":"5FFyP5qQtQKp"}},{"cell_type":"markdown","source":["![Figure 7.14](https://drive.google.com/uc?export=view&id=1Icmd4a9ueJwVYH_Fy0dSYrFA7LD09KhB)"],"metadata":{"id":"JyDYI75ytU6C"}},{"cell_type":"markdown","source":["> **Figure 7.16. The temporal program of expression from a single-input\n","module.** As the activity of $p_x$ rises, it crosses the thresholds of activity for\n","each promoter ($K_{xa}$, $K_{xb}$, and $K_{xc}$) in order. When the activity of $p_x$ declines, it exhibits a “first in, last out” behavior. Adapted by permission from Macmillan\n","Publishers Ltd: Shen-Orr, S. S., Milo, R., Alon, U. Network motifs in the\n","transcriptional regulation network of *Escherichia coli. Nature Genetics*. 2002.\n","**31**(1):64-8."],"metadata":{"id":"WYXlaGTCr-8Y"}},{"cell_type":"markdown","source":["The equations are identical in form, and as you already know, the production and decay terms determine the steady-state expression level of protein once the genes are induced. The thresholds $K_{xa}$, $K_{xb}$, and $K_{xc}$ determine how long it takes for gene expression to be induced after $p_x$ is expressed. For example, if $K_{xa} < K_{xb}$, then it will take longer for $p_x^*$ to be greater than $K_{xb}$ and so $p_b$ expression won’t be induced until later. Similarly, you can see in Figure 7.16 that if $K_{xa} < K_{xb} < K_{xc}$, expression of these genes will occur in the order $g_a$, then $g_b$, then $g_c$ – just as we saw with the arginine regulon (Figure 7.15)"],"metadata":{"id":"06zrUd8ptqiJ"}},{"cell_type":"markdown","source":["What does it mean biologically for these thresholds to be different from each other?\n","Most likely, the difference is related to the binding affinity of the transcription factor for\n","each gene’s promoter. A promoter with a high-affinity binding site would lead to a low\n","threshold: even at a low concentration of the transcription factor, binding occurs and\n","expression is induced."],"metadata":{"id":"kk3o2X6AuGuC"}},{"cell_type":"markdown","source":["The single-input module yields concurrent expression and has the potential for sequential induction of gene expression. Notice from Figure 7.16, however, that when expression of $p_x$ decays, $g_c$ is the first gene to respond, followed by $g_b$ and finally $g_a$. In other words, de-activation of the system occurs in reverse order: first in, last out."],"metadata":{"id":"gamprDixuTjp"}},{"cell_type":"markdown","source":["## **Section 7.11. Generalization of the feed-forward loop**"],"metadata":{"id":"FFCjoyYauIt6"}},{"cell_type":"markdown","source":["Are there any network structures that would give us **first in, first out** behavior? After all, you might expect that this strategy would be even more efficient in terms of enzyme production; enzymes are produced just in time, and only maintained as long as they are needed."],"metadata":{"id":"V7uCJJkbuVFE"}},{"cell_type":"markdown","source":["As it happens, another motif that was shown to be overrepresented in *E. coli’s* network\n","versus randomized networks can indeed produce “first in, first out” behavior. This motif\n","is essentially a generalization of the feed-forward loop. Whereas the feed-forward loops that we considered in Figure 7.3 had only one target gene, in the generalized feed- forward loop, two transcription factors – one under the transcriptional control of the other – control the expression of many target genes. In this sense, the generalized or multi- gene feed-forward loop may also be conceptualized as a hybrid between the feed-forward loop and the single-input module."],"metadata":{"id":"W9wpSWp0udEf"}},{"cell_type":"markdown","source":["## <u> **Practice Problem 7.2** </u>"],"metadata":{"id":"KJoDgj4tuio1"}},{"cell_type":"markdown","source":["*Using the ODE approach we adopted for the single-input motif, write the equations for\n","the multi-gene feed-forward loop in Figure 7.17, and describe the conditions for which\n","first in, first out expression of the genes z1 and z2 will be achieved.*"],"metadata":{"id":"u5zmSdiquoJG"}},{"cell_type":"markdown","source":["![Figure 7.14](https://drive.google.com/uc?export=view&id=12qntOFoyzNGpu97zAhE2tbWo1D3AAaKe)\n"],"metadata":{"id":"u9XPmwqEurjW"}},{"cell_type":"markdown","source":["> **Figure 7.17. A two-node feed-forward loop with two gene targets.** The\n","transcription factors that control expression are common to both targets."],"metadata":{"id":"zLU_1jWeuuhr"}},{"cell_type":"markdown","source":["**Solution:** The equations appear below. Assume that $s_x$ and $s_y$ are sufficiently available such that $p_x = p_x^*$ and $p_y = p_y^*$.\n"],"metadata":{"id":"57dQiLMtvHJl"}},{"cell_type":"markdown","source":["\n","$\n","\\frac{d[p_x]}{dt} = k_{x_{\\text{prod}}} - k_{x_{\\text{deg}}} [p_x]\n","$\n","\n","$\n","\\frac{d[p_y]}{dt} = k_{y_{\\text{prod}}} \\theta \\left( \\left[ p_x^* \\right] > K_{xy} \\right) - k_{y_{\\text{deg}}} [p_y]\n","$\n","\n","$\n","\\frac{d[p_{z1}]}{dt} = k_{z1_{\\text{prod}}} \\left[ \\theta \\left( \\left[ p_x^* \\right] > K_{xz1} \\right) \\text{ OR } \\theta \\left( \\left[ p_y^* \\right] > K_{yz1} \\right) \\right] - k_{z1_{\\text{deg}}} [p_{z1}]\n","$\n","\n","$\n","\\frac{d[p_{z2}]}{dt} = k_{z2_{\\text{prod}}} \\left[ \\theta \\left( \\left[ p_x^* \\right] > K_{xz2} \\right) \\text{ OR } \\theta \\left( \\left[ p_y^* \\right] > K_{yz2} \\right) \\right] - k_{z2_{\\text{deg}}} [p_{z2}]\n","$"],"metadata":{"id":"37ikt09WvNWW"}},{"cell_type":"markdown","source":["From our work in Section 7.10, you should already have an intuition that first in, first out behavior depends on the threshold $K$ parameters. If $K_{xz1} < K_{xz2}$, then $p_{z1}$ will begin to be expressed earlier than $p_{z2}$ (Figure 7.18)."],"metadata":{"id":"ptdfBV4Ruw1v"}},{"cell_type":"markdown","source":["![Figure 7.14](https://drive.google.com/uc?export=view&id=1th9lYEs1Ghhh9LffUcGUFnU40XAhiCFI)"],"metadata":{"id":"36nSnGjgvcZ2"}},{"cell_type":"markdown","source":["> **Figure 7.18. The output of the multi-node feed-forward loop in Figure\n","7.17, which exhibits “first in, first out” dynamics.** Due to the OR\n","relationship between the transcription factors, the expression of $p_x$, which\n","precedes $p_y$ expression, controls the initial induction of $p_{z1}$ (black) and $p_{z2}$\n","(red). $p_{z1}$ is expressed before $p_{z2}$ because Kxz1 < Kxz2. The decrease in $p_y$\n","expression occurs after $p_x$ decreases, so $p_y$ controls the decreases in $p_{z1}$ and\n","$p_{z2}$. Here, $K_{yz1} > K_{yz2}$, and so $p_{z1}$ expression decreases before $p_{z2}$ expression\n","does. $p_{z1}$ levels go up first, then $p_{z2}$ levels rise; since the levels of $p_x$ and $p_y$\n","decrease, once again $p_{z1}$ levels decrease before $p_{z2}$ levels do. Modified from\n","Alon, U. *An Introduction to Systems Biology: Design Principles of\n","Biological Circuits.* Chapman & Hall/CRC, 2007. Reproduced with\n","permission of TAYLOR & FRANCIS GROUP LLC in the format Republish\n","in a book via Co$p_y$right Clearance Center."],"metadata":{"id":"x1pNquBCveGd"}},{"cell_type":"markdown","source":["Look at the right side of Figure 7.18 to find the requirements for “first out,” or in our case, $p_{z1}$’s expression dropping before that of $p_{z2}$. Notice that the removal of $p_x$ from the system does not affect $p_{z1}$ or $p_{z2}$ due to the OR relationship that is encoded into the expression of both of the corresponding genes. Instead, it is the removal of $p_y$ after $p_x$ has already begun to be lost (and passed the $K_{xy}$ threshold) that determines when $p_{z1}$ and $p_{z2}$ begin to be removed from the system. In this case, the requirement for $p_{z1}$ to be removed first is reflected by the fact that $K_{yz1} > K_{yz2}$. Taken together, these parameters lead to first in, first out expression."],"metadata":{"id":"Z82rJGpmv__9"}},{"cell_type":"markdown","source":["## **Section 7.12. An example of a multi-gene feed-forward loop: flagellar biosynthesis in *E.coli***"],"metadata":{"id":"tZ3nYh4QwMXY"}},{"cell_type":"markdown","source":["Now we’ve identified the network structure (multi-node feed-forward loop) and the\n","conditions (OR interactions at the promoter, and certain relationships between the\n","threshold parameters) that lead to first in, last out expression of target genes. But is this\n","kind of behavior ever exhibited by live *E. coli*?"],"metadata":{"id":"xLLdiYFxwZPf"}},{"cell_type":"markdown","source":["An impressive real example that produces similar dynamics occurs during construction of\n","the bacterial flagellum, the long “tail” that *E. coli* uses to propel itself through its liquid\n","environment. When *E. coli* that grow flagella (some lab strains don’t) are placed in an\n","environment without much food, they grow flagella so that they can look for more\n","nutrient-rich environments. This system has been very well characterized; we know\n","which genes encode all of the parts of the flagellum (Figure 7.19A). As we saw with the\n","arginine biosynthesis single-input module (Figure 7.15), the flagellar genes are expressed\n","in the order in which they are required (Figure 7.19B). However, in this case, two\n","transcription factors regulate the expression of the flagellar biosynthesis genes: FlhDC\n","and FliA. Furthermore, as you saw in Section 7.8, FliA expression is also regulated by\n","FlhDC. The resulting multi-node feed-forward loop is shown in Figure 7.19C."],"metadata":{"id":"zRxFhAxSwZKw"}},{"cell_type":"markdown","source":["![Figure 7.14](https://drive.google.com/uc?export=view&id=1HipdIWX4TbY-KbIQR-a-vWppKEL1vQv6)"],"metadata":{"id":"D1cc4nqdwg37"}},{"cell_type":"markdown","source":["> **Figure 7.19. Regulation of the flagellar biosynthesis genes via a multi-node\n","feed-forward loop driven by FlhCD and FliA.** (A) The genes are linked to the\n","part of the flagellum that they encode. (B) Time course of gene expression for all of\n","the genes involved in the circuit. Promoter activity was monitored by transcriptional\n","fusions to GFP. As in Figure 7.16, locating the inflection point facilitates\n","visualization of the order of gene expression. (C) Schematic of the regulation of the system. A “+” indicates that only the first gene in the operon is listed. Modified\n","from Kalir, S., McClure, J., Pabbaraju, K., Southward, C., Ronen, M., Leibler, S.,\n","Surette, M. G., Alon, U. Ordering genes in a flagella pathway by analysis of\n","expression kinetics from living bacteria. Science. 2001. 292: 2080-3. Reprinted\n","with permission from AAAS."],"metadata":{"id":"Aiym6UZ3wioJ"}},{"cell_type":"markdown","source":["Alon’s group was particularly interested in the structure of this multi-gene feed-forward\n","loop and subjected it to intense experimental scrutiny (see Kalir et al. 2001 in\n","Recommended Reading). They verified that FliA regulated the expression of even the\n","early-expressed genes (previously thought to be only under FlhDC control), probed the\n","nature of the interaction between FlhDC and FliA at the promoter regions (a SUM\n","relationship, which is very similar to, but slightly more complex than, an OR\n","relationship), and determined the dynamics of activation for FlhDC, FliA, and the target\n","transcription units."],"metadata":{"id":"4yDVLz17wpcS"}},{"cell_type":"markdown","source":["Those dynamics can be seen in Figure 7.20. The activity of each transcription factor\n","(corresponding roughly to ${p_x}^*$ and ${p_y}^*$ in Practice Problem 7.2) is measured by the\n","expression of GFP, whose gene is fused to a protein that is under the sole control of either\n","FlhDC or FliA (Figure 7.20A). The other seven promoter activities (Figure 7.20B) are\n","taken from the endogenous flagellar biosynthesis promoters, also linked to GFP\n","expression."],"metadata":{"id":"iZFz9B0kwqhR"}},{"cell_type":"markdown","source":["![Figure 7.14](https://drive.google.com/uc?export=view&id=1YPNC5KGz0ajrV6uhY1wVCxxdDzaN2Xza)"],"metadata":{"id":"tcMpLioFw7VE"}},{"cell_type":"markdown","source":["> **Figure 7.20. Dynamics from the flagellar biosynthesis network, determined\n","experimentally.** See Figure 7.19C for the regulatory schematic of this system. (A)\n","Black line, the abundance of the fluorescent reporter fused to a mutated *fliL* promoter\n","that does not bind FliA reflects the activity of the *flhDC* promoter. Grey line,\n","fluorescent protein abundance when the reporter is fused to a promoter that is only\n","responsive to FliA. (B) The activities of the various promoters change as the network\n","shifts from FlhDC-dominated to FliA-dominated. OD stands for optical density, a\n","measure of how much light passes through a bacterial culture; the more bacteria that\n","have grown over time, the denser the solution. Promoter activity is normalized by\n","dividing the measured promoter fluorescence by the OD. Normalization controls for\n","the fact that fluorescence increases not only as a result of promoter activity, but also\n","as a by-product of cell growth. Kalir, S. and Alon, U. Using a quantitative blueprint\n","to reprogram the dynamics of the flagella gene network. *Cell.* 2004. **117**:713-20.\n","Reprinted with minor modifications with permission from Elsevier."],"metadata":{"id":"AnVW7_vgw-AL"}},{"cell_type":"markdown","source":["Notice that FliA activity increases sharply after FlhDC reaches its full activity. This\n","makes sense, because FliA expression depends on FlhDC. Now look at the endogenous\n","promoter readouts in Figure 7.20B. During the first phase of expression, when FlhDC is\n","the predominantly active transcription factor, the promoter activities differ within an\n","order of magnitude. We already saw in Figure 7.19B that these genes are expressed in a\n","specific order. However, once FliA becomes the predominant transcription factor, some\n","promoter activities increase and others decrease, such that all of the promoters have the\n","same activity (Figure 7.20B)."],"metadata":{"id":"uWC4LC7zxJnd"}},{"cell_type":"markdown","source":["In other words, these activities do not quite lead to “first-in, first-out” expression\n","patterns; it’s more like “first-in” followed by a constant rate of maintenance across the\n","transcription units. Such a pattern is interesting in its own right, and would not be\n","possible with a single-input module. In any event, the coordination of two transcription\n","factors can produce dynamics that are quite complex!"],"metadata":{"id":"wzzrpqprxKLB"}},{"cell_type":"markdown","source":["## **Section 7.13. Other regulatory motifs**"],"metadata":{"id":"bWQEm81UxLeJ"}},{"cell_type":"markdown","source":["Two other motifs were found to be overrepresented in the *E. coli* and yeast transcriptional\n","regulatory networks: the **bi-fan motif** (Figure 7.21A) and the **dense overlapping region** (Figure 7.21B). The bi-fan motif was the only overrepresented four-node motif; it occurs when two transcription factors each regulate the same two target genes. The dense\n","overlapping region is a generalization of the four-node motif in which a set of\n","transcription factors regulates a common set of target genes. For this motif, it’s not\n","strictly necessary that all of the transcription factors regulate all of the target genes, only\n","that all transcription factors regulate or are regulated by more than one factor or gene and all transcription factors are connected. The properties of these last two motifs haven’t been investigated in detail but I mention them here for completeness. Maybe\n","characterizing them is work for you to do someday!"],"metadata":{"id":"KTer5InrxO21"}},{"cell_type":"markdown","source":["![Figure 7.14](https://drive.google.com/uc?export=view&id=1QVJLu5ofXOwVCog8F9qriEnZ9lYtXcvL)"],"metadata":{"id":"90kD5Zyhxbwq"}},{"cell_type":"markdown","source":["> **Figure 7.21. The last two motifs found to be overrepresented in *E. coli* and yeast.**\n","(A) A bi-fan motif. (B) A dense overlapping region."],"metadata":{"id":"zeMoNOsfxeYc"}},{"cell_type":"markdown","source":["Now we’ve covered essentially all of the motifs that have been identified in a\n","transcriptional regulatory network! I hope you’ve come to appreciate how dynamics can\n","change with network structure, and that you’ve obtained a sense of how you could put\n","many of these motif models together to begin assembling networks representative of the\n","whole cell."],"metadata":{"id":"LEsHrYR4xk1W"}},{"cell_type":"markdown","source":["## **Chapter Summary**"],"metadata":{"id":"MrCFNVBpxlnD"}},{"cell_type":"markdown","source":["To move toward whole-cell modeling, we need to progress from the simple regulatory\n","circuits we considered in the earlier chapters of the book to more complex regulatory\n","circuits. Motifs that occur significantly more often in the *E. coli* and yeast transcriptional\n","regulatory networks (as compared to randomly generated networks) have been identified:\n","autoregulation, the feed-forward loop, the single-input motif, multi-gene feed-forward\n","loops, bi-fans, and dense overlapping regions."],"metadata":{"id":"ZEht5P9kxnSG"}},{"cell_type":"markdown","source":["Feed-forward loops are motifs in which a gene’s expression is regulated by two\n","transcription factors, one of which is also regulated by the other. Such loops can be\n","internally consistent, or coherent, if both arms of the loop exert the same type of\n","regulation (are either both positive or both negative). Transcription factors can interact in\n","various ways to regulate gene expression; we considered AND and OR relationships, and\n","mentioned the slightly more complicated SUM relationship."],"metadata":{"id":"fvZo80R7xuP7"}},{"cell_type":"markdown","source":["We modeled the dynamics of three instances of feed-forward loops, primarily using the\n","Boolean approaches developed in Chapter 2. First, we modeled a coherent feed-forward\n","loop in which all regulation was positive and there was an AND relationship between the\n","transcription factors. Expression of the target gene was delayed (relative to the\n","expression of the regulated transcription factor) during induction, but not during the\n","repression of expression. A similar feed-forward loop (only differing by an OR\n","interaction) exhibited a delay in expression change during repression but not induction.\n","We used the analogy of controlling an elevator door’s opening and closing to suggest\n","how such direction-sensitive delays could be useful to the cell (such utility has not been\n","established to date). Using ODEs, we derived an equation for the delay time, and showed\n","that these feed-forward loops also exhibit robustness to short perturbations. The third instance of a feed forward loop was incoherent: one arm regulated target gene expression\n","positively and the other negatively. The most common type of incoherent feed-forward\n","loop in *E. coli* produces short pulses of target gene expression. Satisfyingly, examples of\n","all three of these feed-forward loops have been investigated in *E. coli*, and exhibit the dynamics predicted by theory."],"metadata":{"id":"zJ0IBapzxuvs"}},{"cell_type":"markdown","source":["The single-input motif has a single transcription factor that regulates the expression of\n","many transcription units, often including its own gene. Using an ODE-based approach,\n","we demonstrated that such networks can exhibit just-in-time dynamics in which gene\n","expression is induced sequentially rather than simultaneously. Just-in-time expression\n","dynamics have been observed in *E. coli* in metabolic biosynthesis pathways, where it\n","could be advantageous to produce the next metabolite in the pathway at sufficient\n","concentrations before expressing the enzyme that can bind and convert it to something\n","else."],"metadata":{"id":"LwGNm6lNx3PN"}},{"cell_type":"markdown","source":["According to our simple model, the first protein expressed in the single-input motif is\n","\n","destined to be the last protein remaining after repression of expression. This first-in, last-\n","out expression may be useful in some cases, but intuition suggests that first-in, first-out\n","\n","dynamics, which can be displayed by multi-node feed-forward loops under certain\n","conditions, may be more useful to the cell. Although first-in, first-out dynamics have not\n","been observed to my knowledge, we did consider a related case in the *E. coli* flagellar\n","biosynthesis transcriptional regulatory network."],"metadata":{"id":"vfQ2lzqHx6LN"}},{"cell_type":"markdown","source":["**Recommended Reading**\n","\n","\n","* Alon, U. An Introduction to Systems Biology: Design Principles of Biological Circuits. Chapman & Hall/CRC, 2007.\n","\n","* Baltimore, D. Our genome unveiled. 2001. Nature. 409:814-16.\n","\n","* Davidson, E. H. The Regulatory Genome: Gene Regulatory Networks in Development\n","and Evolution. Academic Press, 2006.\n","\n","* Human Genome Project Information,\n","http://web.ornl.gov/sci/techresources/Human_Genome/home.shtml. This website, from\n","the Oak Ridge National Laboratory and the Department of Energy, contains information\n","about the Human Genome Project and associated issues spanning bioethics, medicine,\n","education, and progress after the publication of the human genome sequence.\n","\n","* Kalir, S., Alon, U. Using a quantitative blueprint to reprogram the dynamics of the\n","flagella gene network. Cell. 2004. 117:713-20.\n","\n","* Kalir, S., Mangan, S., Alon, U. A coherent feed-forward loop with a SUM input function\n","prolongs flagella expression in Escherichia coli. Molecular Systems Biology. Epub 2005\n","Mar 29. doi: 10.1038/msb4100010\n","\n","* Kalir, S., McClure, J., Pabbaraju, K., Southward, C., Ronen, M., Leibler, S., Surette, M.\n","G., Alon, U. Ordering genes in a flagella pathway by analysis of expression kinetics\n","from living bacteria. Science. 2001. 292: 2080-3.\n","\n","* Mangan, S., Itzkovitz, S., Zaslaver, A., Alon, U. The incoherent feed-forward loop\n","accelerates the response-time of the gal system of Escherichia coli. Journal of Molecular\n","Biology. 2006. 356, 1073-1081.\n","\n","* Mangan, S., Zaslaver, A., Alon, U. The coherent feedforward loop serves as a sign-\n","sensitive delay element in transcription networks. Journal of Molecular Biology. 2003.334(2): 197-204.\n","\n","* Rosenfeld, N., Elowitz, M. B., Alon, U. Negative autoregulation speeds the response\n","times of transcription networks. Journal of Molecular Biology. 2002. 323(5): 785-93.\n","\n","* Shen-Orr, S. S., Milo, R., Alon, U. Network motifs in the transcriptional regulation\n","network of Escherichia coli. Nature Genetics. 2002. 31(1):64-8.\n","\n","* Zaslaver, A., Mayo, A. E., Rosenberg, R., Bashkin, P., Sberro, H., Tsalyuk, M., Surette,\n","M. G., Alon, U. Just-in-time transcription program in metabolic pathways. Nature\n","Genetics. 2004. 36(5): 486-91."],"metadata":{"id":"cUTBCGRpx9LG"}},{"cell_type":"markdown","source":["## **Problems**"],"metadata":{"id":"-0P04CzHydzL"}},{"cell_type":"markdown","source":["### **Problem 7.1. A coherent feed-forward loop with repression**"],"metadata":{"id":"Fe-Ir9qgygBJ"}},{"cell_type":"markdown","source":["In Section 7.4, we examined the coherent type-1 feed-forward loop, which is the most abundant type of feed-forward loop in biological networks. Here, let’s use the notation we developed in Section 7.4 to look at a different type of coherent feed-forward loop in which ${p_x}^*$ induces $p_y$ expression, but both ${p_x}^*$ and $p_y^*$ repress $p_z$ expression. Lumping expression and activation of $p_x$ and $p_y$ into single equations, the Boolean rules that describe this circuit are:\n"],"metadata":{"id":"-QgNKk6Ayj2O"}},{"cell_type":"markdown","source":["\n","$\\text{Activation}_{px} = IF (s_x)$\n","\n","$\\text{Activation}_{py} = $IF $({p_x}^*)$ AND $(s_y)$\n","\n","$\\text{Expression}_{pz}$ = IF NOT $({p_x}^)$ AND NOT $({p_y}^)$\n","\n","${p_x}^* = \\text{IF} (\\text{Activation}_{px}) \\text{AFTER SOME TIME}$\n","\n","$p_y^* = \\text{IF} (\\text{Activation}_{py}) \\text{AFTER SOME TIME}$\n","\n","$p_z = \\text{IF} (\\text{Expression}_{pz}) \\text{AFTER SOME TIME}$"],"metadata":{"id":"fxdYkBwl0BUO"}},{"cell_type":"markdown","source":["**(a)** Draw a diagram of the feed-forward loop for this system."],"metadata":{"id":"DMt8Z32q0Bxq"}},{"cell_type":"markdown","source":["**(b)** Construct a state diagram for this motif, for the case in which $s_x = 1$ and $s_y = 1$. Start from $p_x^* = p_y^* = p_z = 0$ and fill in the rows in your matrix until you reach a stable state."],"metadata":{"id":"QNdQfGiv0Fhw"}},{"cell_type":"markdown","source":["**(c)** What happens when you remove $s_x$ from the system in (b)? Add another column to your state diagram to answer this question."],"metadata":{"id":"T-p_kvGn0Htn"}},{"cell_type":"markdown","source":["**(d)** Use your matrix from (b) to graph the time dynamics of $p_x^$, $p_y^$, and $p_z$ in the feed-forward loop. Begin with the addition of $s_x$ and $s_y$, graph the progression of the system until it reaches a steady state, then remove $s_x$ and follow the system to the new steady state. Are there delays in the dynamics of $p_z$?\n"],"metadata":{"id":"3fRFxaeb0KG2"}},{"cell_type":"markdown","source":["**(e)** Repeat your state-diagram analysis from (b), but change the motif so that the equation for $p_z^*$ contains an OR instead of an AND gate. Are the dynamics of $p_z$ delayed now?\n","\n","\n","\n","\n","\n","\n"],"metadata":{"id":"HV322Fpk0NPN"}},{"cell_type":"markdown","source":["### **Problem 7.2: A synthetic edge-detection system**"],"metadata":{"id":"uA2wR1nP0PQa"}},{"cell_type":"markdown","source":["Here, we will analyze a synthetic transcriptional regulatory circuit designed to detect the\n","edges of an image of light and produce a dark pigment in response (inspired by an actual\n","circuit described in Tabor, J. J., Salis, H. M., Simpson, Z. B., et. al. A synthetic genetic\n","edge detection program. Cell. 2009. **137**:1272-81). The circuit consists of two signals\n","as inputs: Darkness and Light. There are three proteins, px, py, and pz, all of which also\n","have active forms. The Boolean rules that describe the system are:"],"metadata":{"id":"LWBGNwt-0Ts6"}},{"cell_type":"markdown","source":["$\\text{Activation}_{p_x^*} = \\text{IF (Darkness)}$\n","\n","$\\text{Activation}_{p_y^*} = \\text{IF (}p_x^* \\text{) AND (Light)}$\n","\n","$\\text{Expression}_{p_z} = \\text{IF } p_x^* \\text{ AND } p_y^*$\n","\n","$p_x^* = \\text{IF (Activation}_{p_x} \\text{) AFTER SOME TIME}$\n","\n","$p_y^* = \\text{IF (Activation}_{p_y} \\text{) AFTER SOME TIME}$\n","\n","$p_z = \\text{IF (Expression}_{p_z} \\text{) AFTER SOME TIME}$\n","\n","$\\text{Pigment} = \\text{IF (} p_z \\text{)}$"],"metadata":{"id":"RsDALPX90XWW"}},{"cell_type":"markdown","source":["\n","**(a)** Draw a diagram of the feed-forward loop in this system.  \n"],"metadata":{"id":"tXhWoLL80y4Z"}},{"cell_type":"markdown","source":["**(b)** Draw the state matrix for the system. Note that in this case, the system must have at least one of the signals (don’t make a column for $s_x = s_y = 0$). Circle any stable states."],"metadata":{"id":"qhTkP-Fx03MJ"}},{"cell_type":"markdown","source":["**(c)** For the schematic below, indicate which numbered regions would produce Pigment. Justify your answer."],"metadata":{"id":"LbDF-H2004qb"}},{"cell_type":"markdown","source":["![Figure 7.14](https://drive.google.com/uc?export=view&id=1EDcFnMD9vjtuFdBM4XCnVqy9trvHHo1a)"],"metadata":{"id":"0EpbQQSo05_T"}},{"cell_type":"markdown","source":["### **Problem 7.3: Advanced analysis of an incoherent feed-forward loop**"],"metadata":{"id":"YVK6HE18091J"}},{"cell_type":"markdown","source":["Certainly! Here's the problem with the appropriate LaTeX formatting for subscripts using dollar signs and the 'Activation' within the \\text brackets:\n"],"metadata":{"id":"i7AbXi8Z1Cll"}},{"cell_type":"markdown","source":["\n","Let's examine one of the most regulatory common motifs, the incoherent feed-forward loop shown below. Your goal is to analyze this system, first using ODEs (Chapter 3) and then using a stochastic simulation (Chapter 6). We will assume that $p_x$ and $p_y$ are always active (signals $s_x$ and $s_y$ are always present). We will also assume that the expression of $p_x$ depends on $s_x$ such that production rate = max production rate $\\cdot s_x = \\beta_x \\cdot s_x$. The Boolean rules for this system are:"],"metadata":{"id":"RrNDM4k31yR8"}},{"cell_type":"markdown","source":["\n","$$\n","p_x = \\text{IF } s_x > 0\n","$$\n","\n","$$\n","p_y = \\text{IF } p_x > K_{xy}\n","$$\n","\n","$$\n","p_z = \\text{IF } p_x > K_{xz} \\text{ AND NOT (} p_y > K_{yz} \\text{)}\n","$$\n","\n","Constants are defined as:\n","$$\n","\\beta_x = \\text{maximal production rate of } p_x \\\\\n","\\beta_y = \\text{maximal production rate of } p_y \\\\\n","\\beta_z = \\text{maximal production rate of } p_z \\\\\n","\\alpha_x = \\text{rate of dilution/degradation of } p_x \\\\\n","\\alpha_y = \\text{rate of dilution/degradation of } p_y \\\\\n","\\alpha_z = \\text{rate of dilution/degradation of } p_z\n","$$\n","\n","Assume that you start with no $p_x$, $p_y$, or $p_z$."],"metadata":{"id":"-iTIqfAJ11fW"}},{"cell_type":"markdown","source":["\n","**(a)** Write an ODE for the change in $[p_x]$ over time."],"metadata":{"id":"qSFv5CMT17Zv"}},{"cell_type":"markdown","source":["**(b)** Write an ODE for the change in $[p_y]$ over time."],"metadata":{"id":"R0_VVIJ32ItF"}},{"cell_type":"markdown","source":["\n","**(c)** Assume that $s_x$ is always present at a maximum value of 1, and that the levels of $p_x$ are higher than the threshold for $p_y$ expression ($[p_x] > K_{xy}$). Write the analytical solution for $[p_y](t)$ for the ODE you wrote in (c)."],"metadata":{"id":"oV-NyNj42Kye"}},{"cell_type":"markdown","source":["\n","**(d)** Assume that $[p_x] > K_{xy}$ and $[p_x] > K_{xz}$, but initially $[p_y] < K_{yz}$. Write the ODE for $[p_z]$ and solve it analytically for $[p_z](t)$."],"metadata":{"id":"GmAf_y8B2M2h"}},{"cell_type":"markdown","source":["**(e)** Continue to assume the same level of $p_x$ ($[p_x] > K_{xy}$ and $[p_x] > K_{xz}$), but the level of $p_y$ is now high enough to stop the expression of $p_z$ ($[p_y] > K_{yz}$). Write an ODE for the levels of $p_z$ over time for this case."],"metadata":{"id":"iCU74uo62O1i"}},{"cell_type":"markdown","source":["**(f)** Use `Python` and all of the expressions above to plot the levels of $p_x$, $p_y$, and $p_z$ over time. Use the following information:\n","\n","$$\n","s_x = 0 \\text{ for } -1 \\leq t < 0 \\\\\n","s_x = 1 \\text{ for } 0 \\leq t < 10\n","$$\n","\n","Constants:\n","$$\n","\\beta_x = 1 \\text{ mM/h} \\\\\n","\\beta_y = 1 \\text{ mM/h} \\\\\n","\\beta_z = 1 \\text{ mM/h} \\\\\n","\\alpha_x = 1 \\text{ /h} \\\\\n","\\alpha_y = 1 \\text{ /h} \\\\\n","\\alpha_z = 1 \\text{ /h} \\\\\n","K_{xz} = 0.4 \\text{ mM} \\\\\n","K_{xy} = 0.4 \\text{ mM} \\\\\n","K_{yz} = 0.5 \\text{ mM}\n","$$\n","\n","Initially, $p_x = p_y = p_z = 0$ mM."],"metadata":{"id":"iDK0F6-C2RDO"}},{"cell_type":"markdown","source":["**(g)** Previously, we assumed that the signal $s_x$ was available for full activation. Now, let’s consider sub-maximal signaling by $s_x$. Using the rate constants and activation thresholds in (f), plot the dynamics of $p_x$ for $s_x = [0.1:0.1:1]$. What behavior do you see in the response of $p_z$ to different levels of $s_x$? What function might this behavior serve in a living cell?"],"metadata":{"id":"2ubHKzv-2aB9"}},{"cell_type":"markdown","source":["\n","**(h)** Increase $K_{xy}$ to 0.8 and run your simulation again for $s_x = [0.1:0.1:1]$. Now how does $p_z$ expression behave in response to various levels of $s_x$? How might this type of regulation be useful for biological systems?"],"metadata":{"id":"4TMwfVxn2b0U"}},{"cell_type":"markdown","source":["\n","**(i)** Build and fill a state diagram for the network and circle the stable states. Does this analysis agree with your results from (f)? Why or why not?"],"metadata":{"id":"iPK75qTi2de6"}},{"cell_type":"markdown","source":["\n","**(j)** Finally, let’s examine this network using a stochastic simulation. Again, we assume that $p_x$ and $p_y$ will always be active ($s_x$ and $s_y$ are abundant). The reactions are:\n","\n","$$\n","\\text{px expression: } DNA_x \\rightarrow DNA_x + p_x \\\\\n","\\text{py expression: } DNA_y + p_x \\rightarrow DNA_y + p_x + p_y \\\\\n","\\text{pz expression: } DNA_z + p_x \\rightarrow DNA_z + p_x + p_z \\\\\n","\\text{py:DNA association: } DNA_z + p_y \\rightarrow DNA_z, p_y \\\\\n","\\text{py:DNA dissociation: } DNA_z, p_y \\rightarrow DNA_z + p_y \\\\\n","\\text{px degradation: } p_x \\rightarrow 0 \\\\\n","\\text{py degradation: } p_y \\rightarrow 0 \\\\\n","\\text{pz degradation: } p_z \\rightarrow 0\n","$$\n","\n","Let the stochastic rate constants be:\n","$$\n","c_{px\\text{Expression}} = 0.8/(\\text{molecules } DNA_x \\cdot \\text{ min}) \\\\\n","c_{py\\text{Expression}} = 0.8/(\\text{molecules } DNA_y \\cdot \\text{ min}) \\\\\n","c_{pz\\text{Expression}} = 0.8/(\\text{molecules } DNA_z \\cdot \\text{ min}) \\\\\n","c_{py\\text{BindingDNAz}} = 0.8/(\\text{molecules } p_y \\cdot \\text{molecules } DNA_z \\cdot \\text{ min}) \\\\\n","c_{py\\text{UnbindingDNAz}} = 0.4/(\\text{molecules } py:DNA_z \\text{ complex} \\cdot \\text{ min}) \\\\\n","c_{px\\text{Degradation}} = 0.001/(\\text{molecules } p_x \\cdot \\text{ min}) \\\\\n","c_{py\\text{Degradation}} = 0.001/(\\text{molecules } p_y \\cdot \\text{ min}) \\\\\n","c_{pz\\text{Degradation}} = 0.001/(\\text{molecules } p_z \\cdot \\text{ min})\n","$$\n","\n","Assume that you begin with no $p_x$, $p_y$, or $p_z$, but you have one molecule each of $DNA_x$, $DNA_y$, and $DNA_z$. Use `Python` to write a Gillespie algorithm and run it for 500,000 steps (you may choose to only store every tenth timepoint). Describe the system’s behavior and interpret it in biological terms.\n","\n","(k) What happens if you change the relative protein activities such that $p_x$ promotes the production of $p_y$ with ten-fold less affinity than it does for $p_z$? (Hint: change a rate constant.) Describe resulting behavior, and compare your results qualitatively with your results from (g), (i), and (j)."],"metadata":{"id":"5wPI2s2V2fZw"}},{"cell_type":"markdown","source":["### **Problem 7.4: A single-input module**"],"metadata":{"id":"NzKpjf7o2hSj"}},{"cell_type":"markdown","source":["Consider the following metabolic pathway, which converts glutamate to ornithine in *E.\n","coli* (slightly modified from Figure 7.15):"],"metadata":{"id":"NY4z2F1T2zgJ"}},{"cell_type":"markdown","source":["![Figure 7.14](https://drive.google.com/uc?export=view&id=1gPTvvDrnbA4svtrSpYFDZbYHhpO4EBVE)"],"metadata":{"id":"cU9ViJtq22P8"}},{"cell_type":"markdown","source":["The genes in this pathway are all regulated by the transcription factor ArgR via the\n","single-input motif depicted at right."],"metadata":{"id":"LanoA56Z246n"}},{"cell_type":"markdown","source":["**(a)** List two main benefits to the cell of regulating all of these genes with ArgR."],"metadata":{"id":"-9ngL_4R27Cc"}},{"cell_type":"markdown","source":["**(b)** Assume that when a signal to activate ArgR is present (sArgR), the change in ArgR\n","production can be defined as:\n","\n","\n","$\n","\\frac{d[\\text{ArgR}^*]}{dt} = \\beta_{\\text{ArgR}} - \\alpha_{\\text{ArgR}} [\\text{ArgR}^*]\n","$\n","\n","The change in all of the other gene products can be defined as:\n","\n","$\n","\\frac{d[\\text{Arg}_i]}{dt} = \\beta_i \\theta \\left( [\\text{ArgR}^*] > K_{\\text{ArgR}, \\text{Arg}_i} \\right) - \\alpha_i [\\text{Arg}_i]\n","$\n","\n","where $i = \\{ \\text{A}, \\text{B}, \\text{C}, \\text{D}, \\text{E} \\}$. Given that $S_{\\text{ArgR}}$ is added at $t = 0$ min and removed at $t = 7$ min, find values for all $K_{\\text{ArgR}, \\text{Arg}_i}$ such that $ \\text{Arg}_i$ will be induced at equal time intervals within the range $[0, t([\\text{ArgR}] = [\\text{ArgR}_{\\text{ss}}]/2)]$, where ss denotes steady state. Note that the gene with the highest threshold parameter will obey $K_{\\text{ArgR}, \\text{Arg}_i} = [\\text{ArgR}_{\\text{ss}}]/2$. Use values of 1/min and 1 $\\mu$M/min for all $\\alpha$ and $\\beta$ terms, respectively."],"metadata":{"id":"9gA_qZ_j29tS"}},{"cell_type":"markdown","source":["**(c)** Use `Python` with the above information to plot the activity of ArgR* as well as\n","the expression of all Arg proteins over the time range [0:14] minutes (choose the\n","most appropriate time step). What can you say about the timing of the decay of\n","the various genes, as compared to the timing of induction? Specifically describe\n","the order and the duration of the decays."],"metadata":{"id":"ZVOXL9kb3XF5"}}]}