{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[]},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["#**Boolean Representations**"],"metadata":{"id":"9xN6eiTYjGyK"}},{"cell_type":"markdown","source":["##***Learning Objectives***\n","\n","*   Construct Boolean logic statements to describe a biological system\n","*   Calculate a state matrix, identifying stable states\n","*   Determine Dynamics of the system based on the state matrix outputs\n","*   Understand how key parameter values affect system dynamics\n","\n"],"metadata":{"id":"W095irTrQbiX"}},{"cell_type":"markdown","source":["We will start our exploration of the negatively autoregulated circuit described in Chapter 1 using a classical Boolean approach that relies on work performed by Rene Thomas at the Universite Libre de Bruxelles in Belgium.  When regulation of gene expression was first discovered, Thomas started exploring how to model it; he began with this approach.  It’s a perfect way to begin building intuition!"],"metadata":{"id":"I6ORlQjvRTDe"}},{"cell_type":"markdown","source":["##***Section 2.1: Boolean Logic and Rules***"],"metadata":{"id":"O63Z-WjART2z"}},{"cell_type":"markdown","source":["The idea behind the Boolean approach is that you identify a number of features of the system that can be represented by one of two states.  For example, mRNA might be considered present in a cell or not, and a protein might be considered active or not.  In many cases, this Boolean representation is a dramatic simplification; in reality, proteins can have intermediate levels of activity and an mRNA can be present at an intermediate level.  However, the beautiful thing about using a Boolean approach to model this system is that you can easily look at every single state of the system, given any inputs and initial conditions.  Such a comprehensive view would be impossible with any of the other approaches we’ll consider in this book.\n","\n","The values that these variables hold are determined by logic rules.  For example, an mRNA may be considered present only if the corresponding gene exists in that system or organism, and the corresponding rule might read:\n","\n","> **<h3> mRNA = IF (Gene) <h3>**\n","\n"],"metadata":{"id":"4tPruCw7Re7y"}},{"cell_type":"markdown","source":["###*Figure 2.1*\n","Figure 2.1 is a drawing of our typical negative autoregulatory circuit from Figure 1.3, represented in a way that will facilitate Boolean logic modeling.  Please note that we will use capital letters to denote variables (such as Protein; for simplicity, we will continue to refer to mRNA-associated variables as “mRNA”) and lower-case letters to denote the physical concepts associated with those variables (mRNA and protein).  For Boolean analyses, we won’t italicize these variables for simplicity.  This extra notation is intended to help you keep all our terms straight."],"metadata":{"id":"Z1DlPqqGS84b"}},{"cell_type":"markdown","source":["![Figure 2.1](https://drive.google.com/uc?export=view&id=1dUNLmMWBkjZhAwVm-eyizqSY3B-cuh7j)"],"metadata":{"id":"e-wsU68MS-fQ"}},{"cell_type":"markdown","source":["> **Figure 2.1.  A Boolean model of our typical negative autoregulatory circuit.** Just as we saw in Figure 1.3, our system encapsulates a gene, transcription into mRNA, translation into protein, activation of the protein by an external factor, and transcriptional repression (blunt arrow) following DNA-protein interaction.  For simplicity, this Boolean model represents translation and activation as a single process called “translation.”"],"metadata":{"id":"Xvq4tCfYTSoZ"}},{"cell_type":"markdown","source":["Just as before, a gene is transcribed to mRNA and then translated into a protein that is activated in the presence of an activator.  Imagine, for example, a small molecule such as oxygen that binds to the inactive protein to activate it (in Figure 2.1, for simplicity we have lumped the translation and activation processes into one process called “translation”).  The active protein then represses further transcription of the gene.\n","\n","We are now going to define several variables that describe the system.  There are entities, such as the Gene, Activator, mRNA, and Protein, and processes, such as Transcription and Translation.  Each of these variables can have a value of either 0 (absent, off, inactive) or 1 (present, on, active), and our model will keep track of all these values.  Furthermore, the presence of the Gene and Activator can be manipulated fairly easily experimentally by knocking out the gene or adding/removing the activator from the cell’s environment.  We can therefore consider them as inputs and set their values, as opposed to writing out a Boolean rule.\n","\n","The Boolean rules for the processes in Figure 2.1 can be written as follows:\n","\n","> **<h3>Transcription = IF (Gene) AND NOT (Protein)<h3>**\n","\n","> **<h3>Translation = IF (mRNA) AND (Activator) <h3>**\n","\n","\n","In the transcription rule, transcription occurs if you have a gene and you don’t have a protein; that’s a repression-based system.  Translation takes place if there’s both an mRNA and an activator.  Importantly, the Boolean value of 0 or 1 does not necessarily have to correspond to a **zero concentration** value, for instance in cases requiring a significantly high concentration of protein for effective regulation.\n"],"metadata":{"id":"Bq_feQoqUXRy"}},{"cell_type":"markdown","source":["The Transcription and Translation rules are instantaneous, meaning that evaluation of their output depends only on the state of the other variables in the system at the same instant in time.  If the mRNA and activator are sufficiently present such that their model values are each 1, then Translation is initiated and also has a value of 1.\n","\n","In contrast, the Boolean rules for the final two entities are somewhat more interesting:\n","\n","> **<h3> mRNA = IF (Transcription) AFTER SOME TIME**\n","> **<h3> Protein = IF (Translation) AFTER SOME TIME**\n","\n","The values of mRNA and Protein are simply equal to the values for the processes that produce them – but with a time delay.  This simple innovation has the very nice feature of incorporating a **dynamical** component to the model.  The Transcription and Translation variables may be thought of as adding a “memory” of earlier times to the system.\n"],"metadata":{"id":"fif2l81qVENp"}},{"cell_type":"markdown","source":["###*Figure 2.2*\n","In addition, the time delay creates stable and unstable states in our system.  For example, consider Figure 2.2, in which the value of Transcription is changed from 0 to 1 at a certain time point.  What follows will be a time period during which Transcription is equal to 1, but mRNA is still equal to 0.  We can think of the system’s state during this period as unstable because the values of mRNA and Transcription are different from each other, even though our Boolean rule for mRNA suggests that they should be equal.\n","\n","After this time period has passed, mRNA will be equal to 1.  This time period, in biological terms, is the time that it takes for RNA polymerase to locate and transcribe the corresponding gene into mRNA.  At some other time point, transcription will be turned off, leading to a change in Transcription from 1 to 0.  Again, there will be a certain time period required for the mRNA to be depleted from the system; in this case, the delay is associated with decay of the mRNA molecule.  We can make similar statements about the relationship between Translation and Protein."],"metadata":{"id":"beCQWWlZVl7L"}},{"cell_type":"markdown","source":["![Figure 2.2](https://drive.google.com/uc?export=view&id=10qusNZjdleSaTROX5wqgVTcdU7i8RoYN)"],"metadata":{"id":"ScEzgMHXWCam"}},{"cell_type":"markdown","source":["> **Figure 2.2.  Stable and unstable states in the Boolean framework.**  The system is originally in a stable state, where the values of Transcription and mRNA are equal.  At time = 0 (shown schematically with a stopwatch), the value of Transcription increases to 1.  This state is unstable, because the mRNA value is not yet equal to the Transcription value; however, after some time the mRNA value increases to 1, and the result is a stable state."],"metadata":{"id":"xlIjsuC2WRLp"}},{"cell_type":"markdown","source":["##***Section 2.2: State Matrices***"],"metadata":{"id":"mWzdzZv2W3UQ"}},{"cell_type":"markdown","source":["As I noted earlier, the most compelling thing about the Boolean approach is the capacity to consider all states at once.  You can do this by drawing a **state matrix.**  As mentioned earlier, we have two inputs, Gene and Activator.  We also have two processes, Transcription and Translation, which lead to changes in the values of mRNA and Protein, respectively.  Since these last two terms correspond to physical entities, it is convenient to define initial conditions with them.\n","\n","Our state matrix is a chart that holds the values of all of these variables.  Let’s begin to draw one for this system for the trivial case in which neither the gene nor the activator is present.  Intuition would tell you that not much can happen to the circuit without a gene to transcribe, so it’s a good example to become familiar with the approach. In this case, the values of our inputs are both equal to 0.\n","\n","Taking mRNA and Protein as defining the initial conditions, there are four possible starting points for the system: either mRNA or protein molecules are present individually, or both together, or neither.  For each of these starting points and the specific input values, we can calculate what should happen to Transcription and Translation.  For example, if neither mRNA nor protein is present, then:\n","\n","> **<h3>Transcription = IF (Gene) AND NOT (Protein) = 0**\n","> **<h3>Translation = IF (mRNA) AND (Activator) = 0**"],"metadata":{"id":"PcwYvzlDXAw0"}},{"cell_type":"markdown","source":["###*Figure 2.3*\n","Figure 2.3 shows this part of the state diagram.  Notice that the inputs are shown on the top, the mRNA and Protein values are shown at the left, and the values for Transcription and Translation are shown in the lower right.\n","\n","You might think that this case is not interesting because nothing is happening, but just because nothing is happening doesn’t make a system uninteresting (just look at some of those legendary pitchers’ duels in baseball).  In this case there is one interesting feature that I would like you to notice, namely that the system is stable under these conditions.  As we learned earlier, when the values of mRNA and Protein are equal to the values of Transcription and Translation, respectively, then the system is stable – and unless the system is perturbed in some way, we don’t expect any change to occur."],"metadata":{"id":"8ih2V4_GXs_j"}},{"cell_type":"markdown","source":["![Figure 2.3](https://drive.google.com/uc?export=view&id=1eQ00knn5GZs9M2npnTRxi590EJQQFhV4)"],"metadata":{"id":"WE8ItL1jYMZ0"}},{"cell_type":"markdown","source":["> **Figure 2.3.  The state matrix for our negative autoregulatory circuit when neither gene nor protein is present in the system.**  The values for the variables Gene, Activator, Transcription (Trs), Translation (Trl), Protein, and mRNA are shown.  This system is stable under the conditions depicted in the matrix."],"metadata":{"id":"lY6lxtwbYaR3"}},{"cell_type":"markdown","source":["##***Section 2.3: State Transitions***"],"metadata":{"id":"VLH37VTWsxiC"}},{"cell_type":"markdown","source":["Let’s see what happens when we change the initial conditions.  For example, what happens if we start with some protein?  In this case, there still is no gene, activator, or mRNA, so Transcription and Translation are evaluated just as before. This makes sense biologically, because there is no gene from which to transcribe.\n","\n","> **<h3>Transcription = IF (Gene) AND NOT (Protein) = 0**\n","> **<h3>Translation = IF (mRNA) and (Activator) = 0**"],"metadata":{"id":"fCsJNuMMs6K7"}},{"cell_type":"markdown","source":["The corresponding row of the state matrix can then be added as shown in Figure 2.4.  Notice that in this case, the system is no longer stable because the value for Translation (0) does not equal the value for Protein (1).  As a result, after some time – when the existing Protein has been depleted from the system – the Protein value will become 0 and the system will once more be stable.  This transition is depicted as a red arrow in Figure 2.4, and the stable state is highlighted with a red box."],"metadata":{"id":"djVRjdtztlgq"}},{"cell_type":"markdown","source":["### *Figure 2.4*"],"metadata":{"id":"QEuE-MtctLev"}},{"cell_type":"markdown","source":["![Figure 2.4](https://drive.google.com/uc?export=view&id=1D-Rew6JtahOiG8ikKgj3mJaoU0qMJbTy)"],"metadata":{"id":"EqUOwdjetQ6r"}},{"cell_type":"markdown","source":["> **Figure 2.4.  An expansion of the initial state matrix of our typical negative autoregulatory circuit.**  This expansion depicts the change in the system when the initial conditions include the presence of protein.  The updated step appears at the bottom of the figure, and the transition and stable state are highlighted by the red arrow and red box, respectively.  Abbreviations are as in Figure 2.3."],"metadata":{"id":"KBihImy9tcH2"}},{"cell_type":"markdown","source":["### *Figure 2.5*"],"metadata":{"id":"hP0hpk1ttvF1"}},{"cell_type":"markdown","source":["If you continue with the rest of the initial conditions for the (Gene = 0, Activator = 0) input, you will obtain a full column of the matrix, as shown in Figure 2.5.  You can see that there is only one stable state for this system.  If the gene and activator are not present, eventually this system will move toward and remain in a stable state in which no mRNA or protein is present and transcription and translation do not occur.  Notably, there is no more than a single transition from any set of initial conditions to a stable state."],"metadata":{"id":"iHN-p4t4tz8q"}},{"cell_type":"markdown","source":["![Figure 2.5](https://drive.google.com/uc?export=view&id=18lsm33ENdxAV9MJ2t0ND-hgUa6PoIAiH)"],"metadata":{"id":"kfhEGsJJt68h"}},{"cell_type":"markdown","source":["> **Figure 2.5.  Full expansion of the initial state matrix of our typical negative autoregulatory circuit.**  This state matrix has now been fully expanded for the input (Gene = 0, Activator = 0).  Only one stable state occurs for this system. Notations and abbreviations are as in Figure 2.3."],"metadata":{"id":"MH3g719_uFzb"}},{"cell_type":"markdown","source":["## ***Section 2.4: Dynamics***"],"metadata":{"id":"iapkrE2RuZ4c"}},{"cell_type":"markdown","source":["### *Figure 2.6*"],"metadata":{"id":"XvnO1EBKukX2"}},{"cell_type":"markdown","source":["Let’s look at what happens when the value of Activator is changed to 1, a case that is shown in Figure 2.6.  The value of Gene, and correspondingly the value of Transcription – independent of initial conditions – remains at 0.  Translation depends on the presence of both activator and mRNA, so its value will be 1 for the two initial conditions in which mRNA has a value of 1.  As before, you can identify the stable states in this column by looking for rows in which the values of Transcription and Translation are equal to the values of mRNA and Protein.  There is one stable state, the same one that we saw in Figure 2.5 in which transcription and translation do not occur and as a result there is no mRNA or protein in the system.\n","\n","The stable state is therefore identical regardless of the value of Activator (when Gene = 0), but the dynamics are not.  Specifically, if the system begins with mRNA, an intermediate state is attained in which protein is being produced.  This first transition is labeled with a (1) in Figure 2.6.  After some time, however, the protein and mRNA are depleted from the system and because there is no gene and no transcription, these molecules cannot be replaced.  The system then undergoes a second transition, labeled with a (2) in Figure 2.6, to the stable state."],"metadata":{"id":"wibOswTPusId"}},{"cell_type":"markdown","source":["![Figure 2.6](https://drive.google.com/uc?export=view&id=1-wTK4Al7Qy769FrBMyy9TQpQNH8mWaVZ)"],"metadata":{"id":"WJLIWwLXu1NU"}},{"cell_type":"markdown","source":["> **Figure 2.6.  The state matrix for our typical negative autoregulatory circuit when (Gene = 0, Activator = 1).**  Our circuit has only one stable state possible, regardless of the presence of activator.  However, the dynamics differ from the case in which activator is absent (Figure 2.5).  Notations and abbreviations are as in Figure 2.3."],"metadata":{"id":"PRxZR82gu_yQ"}},{"cell_type":"markdown","source":["### *Figure 2.7*\n","You can visualize these transitions more directly by plotting the values of the processes, as well as mRNA and Protein, over time.  The state matrix in Figure 2.6 can be used to create the dynamics plots in Figure 2.7.  Although we don’t define the length of time explicitly on the x-axis (not yet anyway), you can get a sense of the relative timing of events and how the system is behaving."],"metadata":{"id":"KrVVL_ABvOq3"}},{"cell_type":"markdown","source":["![Figure 2.6](https://drive.google.com/uc?export=view&id=1wa_R7fwbkJckQGHgtuGoLVgjy0MffPhz)"],"metadata":{"id":"YW8-dm20val5"}},{"cell_type":"markdown","source":["> **Figure 2.7.  Dynamic plots of the system, with initial conditions of (mRNA = Protein = 1, Gene = 0).**  (A) At time = 0, Transcription is set to 0, and after some time the mRNA decays from the system.  (B) This decay causes the value of Translation to be set to 0, and after some time the protein is lost as well.  The arrows and numbers are for direct comparison with those in Figure 2.6."],"metadata":{"id":"kLr8Bmn6vlwO"}},{"cell_type":"markdown","source":["## **<u>Practice Problem 2.1</u>**"],"metadata":{"id":"mIncJuCJvskZ"}},{"cell_type":"markdown","source":["***Question:*** *Using the approach you’ve just learned, sketch out the column of the state diagram corresponding to a present gene, but no activator.  Find the stable states, map out the trajectories, and interpret your results biologically.  Finally, draw a dynamic plot of the system’s behavior with initial conditions of mRNA = 0, Protein = 1.*"],"metadata":{"id":"izpcvSlGv8Y4"}},{"cell_type":"markdown","source":["**Solution:** See Figure 2.8 for the column of the state matrix and the dynamics plot.  The value of Translation is always 0 because there is no activator, but since there is now a gene, the value of Transcription is 1 when no active protein is present.\n","\n","There is still only one stable state, but it has changed!  Now all of the initial conditions will lead to a state in which there is always some mRNA, but no active protein.  We interpret this biologically to mean that no active protein is produced, and as a result gene transcription is not blocked, and so there will always be some mRNA production (as well as production of the inactive protein, which we have not included explicitly in our model).\n"],"metadata":{"id":"_NRD2Ce6xHcj"}},{"cell_type":"markdown","source":["### *Figure 2.8*"],"metadata":{"id":"dhXZ9mKUxPgR"}},{"cell_type":"markdown","source":["![Figure 2.6](https://drive.google.com/uc?export=view&id=1SLfvuJUEpqy8QLaLyed_xGAG8qxjy3j1)"],"metadata":{"id":"6BEgLbizxSt4"}},{"cell_type":"markdown","source":["> **Figure 2.8.  Solution to Practice Problem 2.1: the state matrix and dynamic plot for our typical negative autoregulatory circuit when (Gene = 1, Activator = 0).**  (A) This state matrix shows that the stable state of our system has changed compared to the stable states in Figures 2.3-2.6.  Notations and abbreviations are as in Figure 2.3.  (B, C) Dynamics plots of system behavior with initial conditions of (mRNA = 0, Protein = 1).\n","\n","Now let’s try the most exciting case."],"metadata":{"id":"Y37Cltl4xbqK"}},{"cell_type":"markdown","source":["## **<u>Practice Problem 2.2</u>**"],"metadata":{"id":"Q406vxoSx6Zm"}},{"cell_type":"markdown","source":["***Question:*** *Beginning with the stable state you calculated in Practice Problem 2.1, determine what happens to the system when you add an activator to the medium.  Calculate the new stable state, if any, map out the trajectories, and explain your results biologically.*"],"metadata":{"id":"RNh24uFyyDXM"}},{"cell_type":"markdown","source":["**Solution:** See Figure 2.9 for a matrix that contains all of the states, including the solution to the problem.  The system begins in the stable state in which transcription is active and produces mRNA, but no active protein is present.\n","\n","You will see from the figure that the system has no stable state.  Instead, when we add activator to the system, the value of the Translation process is evaluated as 1, and so after some time active protein is produced (Transition 1 in Figure 2.9).  Once protein exists, it represses transcription, which is then evaluates to 0, leading to depletion of mRNA after some time (Transition 2).  Without mRNA, protein can no longer be produced and so after more time the value of Protein is also reset to 0 (Transition 3).  However, the absence of active protein means that transcription can take place again, and so after a while mRNA is produced again (Transition 4).\n"],"metadata":{"id":"oqHB4RnZzLBw"}},{"cell_type":"markdown","source":["### *Figure 2.9*"],"metadata":{"id":"DjuDi1lL0Wx5"}},{"cell_type":"markdown","source":["![Figure 2.9](https://drive.google.com/uc?export=view&id=1tNfa39fmLLbLcrsmZiIwH383ZD8Eunkx)"],"metadata":{"id":"QEjN_pgezqlF"}},{"cell_type":"markdown","source":["> **Figure 2.9.  The complete state matrix for our typical negative autoregulatory circuit, including the case in which (Gene = 1, Activator = 1).**  This solution to Practice Problem 2.2 reveals that under these conditions, our circuit has no stable state.  The trajectory specific to (Gene = 1, Activator = 1) is highlighted in red; abbreviations are as in Figure 2.3."],"metadata":{"id":"zSyYyG8Fz3aX"}},{"cell_type":"markdown","source":["Bottom line: this system exhibits oscillation!  It’s pretty fascinating that in this very reduced model with a highly simplified Boolean representation, we are nonetheless able to predict some pretty interesting and complicated dynamics.  We’ve been able to push from a stable state (Figures 2.2-2.8) into an unstable state (Figure 2.9).  The circuit perpetually oscillates when both gene and activator are present, and all this information can be modeled with just this set of simple Boolean rules.\n","\n","Another thing to notice here is that Figure 2.9 contains all of the possible states of this system.  Every combination of input and initial condition values has been evaluated, and all of the results appear on one chart.  You can tell where the stable states are, and you also know how any combination of factors leads to those stable states – or to the oscillating case of (Gene = 1, Activator = 1)."],"metadata":{"id":"5eOMrN7J0D51"}},{"cell_type":"markdown","source":["### *Figure 2.10*"],"metadata":{"id":"Fs5HWcV-0Fld"}},{"cell_type":"markdown","source":["From the state matrix it is relatively easy to draw time courses of the different variables as well.  For example, Figure 2.10 shows the dynamics of mRNA, protein, transcription, and translation over time for the series of transitions highlighted in Figure 2.9.  Again, once the activator is added, Translation is immediately evaluated to 1, followed by Protein transitioning to 1 after some time period.  The presence of protein cuts off mRNA production, and mRNA decays after time, halting translation and leading to depletion of protein and subsequent transcription in a new iteration of the cycle."],"metadata":{"id":"gyExF1xx0JtM"}},{"cell_type":"markdown","source":["![Figure 2.10](https://drive.google.com/uc?export=view&id=1-ULI26FDROA08l1EAh4bx36c3SuE6UUX)"],"metadata":{"id":"5A8m3Lg-0dIy"}},{"cell_type":"markdown","source":["> **Figure 2.10.  The dynamics over time for the (A) mRNA and (B) Protein variables from Figure 2.9.**  The corresponding transitions in the state matrix appear at the top.  mRNA decay and protein dilution are inferred in the absence of transcription (grey dashed line) and translation (black dashed line), respectively."],"metadata":{"id":"-HiJhJbb0fjm"}},{"cell_type":"markdown","source":["## ***Section 2.5: Timescales***"],"metadata":{"id":"GF6dUpRw0vPy"}},{"cell_type":"markdown","source":["One key assumption was made in drawing Figure 2.10: the time periods for mRNA production as well as protein production and depletion were considered to be roughly equal.  In reality, the time periods for mRNA and protein production and depletion can all differ significantly from each other.\n","\n","Let’s consider production first.  How long does it take to make a typical mRNA?  We can use some observations from the literature to make a rough estimate; I’ve compiled some of these numbers in Table 2.1.  A typical RNA polymerase in *E. coli* travels at a rate of nearly 50 nucleotides per second, and a typical gene is approximately 1,000 nucleotides long.  Those **parameters** lead us to an estimate of at least 20 s to make the transcript, and if we add in the time required for RNA polymerase to find the promoter, bind, and eventually finish transcription, a good estimate for mRNA production might be 1-2 min.\n","\n","Protein production turns out to occur on a similar time scale.  Ribosomes attach to free mRNAs one after the other in “polysomes” that look like pearls on a string.  In this way ~10 proteins can be made simultaneously from a single transcript.  The ribosome also travels at ~50 nucleotides (or ~16-17 amino acids produced) per second, which gives us ~20 seconds for **elongation** of a 333 amino-acid peptide from the 1,000 nucleotides-long gene.  Again, when initiation and termination are taken into account, we can say it takes 1-2 min from mRNA existence to protein creation.\n"],"metadata":{"id":"W2UN2ZDB01CR"}},{"cell_type":"markdown","source":["### *Table 2.1*\n"],"metadata":{"id":"mCoeMitt1aMO"}},{"cell_type":"markdown","source":["**Rough estimates of key parameters related to gene expression in *E. coli*.**\n","\n","| Parameter                                    | Estimate        |\n","| -------------------------------------------- | --------------- |\n","| Number of genes in the *E. coli* genome        | ~4,400          |\n","| Average gene length                          | ~1,000 nucleotides |\n","| Time required to produce mRNA from a gene    | ~1-2 min        |\n","| Time required to produce protein from an mRNA| ~1-2 min        |\n","| Half-life of a typical mRNA                  | ~3-8 min        |\n","| Half-life of a typical protein               | ~5-10 h         |\n","| Doubling time during growth in a rich environment | ~30 min     |"],"metadata":{"id":"286cuyZn1qjJ"}},{"cell_type":"markdown","source":["### *Figure 2.11*"],"metadata":{"id":"ENV2CDoY4VxL"}},{"cell_type":"markdown","source":["Now it’s time to consider depletion.  The mRNA decay rates that have been measured in *E. coli* indicate a typical mRNA **half-life** (the time it takes for one-half of the mRNA to decay) of 3-8 min.  The measured protein half-life is substantially longer, probably on the order of several hours on average (although certain proteins are much less stable).\n","\n","mRNA and protein are not only depleted by decay, however.  These concentrations are also reduced by one-half every time that *E. coli* divides.  Concentration loss by cell division is called **dilution**, and has its own time scale – the time scale of cell growth.  For *E. coli* growing under typical lab conditions, we can assume that cell division occurs roughly every 30 min.\n","\n","The division time is significantly longer than the mRNA decay rate, but much shorter than the protein decay rate.  Thus, mRNA depletion will be largely driven by decay, with a half-life of ~5 min, while protein depletion will be mostly due to dilution, with an effective half-life of ~30 min."],"metadata":{"id":"ONeO_bzh4Z73"}},{"cell_type":"markdown","source":["![Figure 2.11](https://drive.google.com/uc?export=view&id=1xBE-gHZcheTYgBU17-YUWXMqzHZZ_20x)\n"],"metadata":{"id":"W0OOu6_b4ljm"}},{"cell_type":"markdown","source":["> **Figure 2.11.  Adding the time-scale information to the dynamics over time shown in Figure 2.10.**  Notice the loss of cyclical behavior within a few *E. coli* generations.  Symbols and notation are as in Figure 2.10."],"metadata":{"id":"a66vVLR14n0J"}},{"cell_type":"markdown","source":["Adding what we know about time scales to our time course plots has a dramatic effect on the predicted behavior of the negative feedback loop (Figure 2.11).  We begin at the same point as in Figure 2.10, with mRNA but no protein, and then add the activator.  Translation is initiated, and we assume it takes on the order of a few minutes to produce enough protein for our Protein variable to equal 1 (transition 1 in Figure 2.9).  Then transcription is repressed, and mRNA decay leads to loss of mRNA from the system within approximately ten more minutes.  Loss of mRNA means that protein will eventually be diluted, but this process is going to take much longer than the buildup of either species, or mRNA decay.\n","\n","As a result, over a short experiment covering a couple of hours, our predicted behavior does not look cyclic at all (although if you plotted for long enough, you would see Protein return to zero and the cycle would continue).  Additionally, our result suggests that if you want to change the dynamic behavior of the system, a critical parameter will be the protein half-life.  In this case, if we reduce the time it takes for protein to be depleted, we get back to the oscillatory case of Figure 2.11.  People have designed proteins with short half-lives for just this reason, as you will see.\n"],"metadata":{"id":"zacRDsDN42Bg"}},{"cell_type":"markdown","source":["## ***Section 2.6: Advantages and Disadvantages of Boolean Analysis***"],"metadata":{"id":"sEByggB_5BXU"}},{"cell_type":"markdown","source":["The Boolean approach that we have described here has some advantages and some disadvantages.  The disadvantages are fairly obvious: the rules and values are gross simplifications of the underlying biology, and so we can’t necessarily rely too much on the results (or predictions) that we obtain.  We will have to be more thorough and read the published literature, conduct experiments, or both, especially if we have a particular biological system in mind.\n","\n","But the main advantages of the Boolean approach – the method’s simplicity, and our ability to consider every single state in a system – are formidable.  All you need to perform the analysis described here is paper and pencil to obtain a significant amount of intuition about the system.  Furthermore, in cases in which not much is known about a system, you can often still guess at Boolean rules and begin to make testable predictions.\n","\n","In the next chapter, we’ll compare this method with ordinary differential equations."],"metadata":{"id":"DdKgJuRp5Jw3"}},{"cell_type":"markdown","source":["## **Chapter Summary**\n","To model our autoregulatory circuit, we first consider the Boolean method, wherein all of the variables can only hold values of 0 or 1.  Equations are written that describe whether or not a biological process occurs, given the values of the inputs and the initial conditions of the system.  Another set of equations, which gives the output conditions, depends on the processes, but only “after some time.”  These equations therefore add a dynamic component to the model.  All of the inputs, processes, and conditions can be represented together in a state matrix, and the state matrix can be used to draw time plots of the system’s dynamics.  The main advantages of the Boolean method are: (1) the equations are simple to write and evaluate; and (2) we can see all of the solutions in a single matrix.  The disadvantages are that a Boolean representation is often too simplistic, and that without some additional information about relative time scales, the method can lead to incomplete, and possibly even misleading, predictions."],"metadata":{"id":"eElIg33S5TZg"}},{"cell_type":"markdown","source":["## **Recommended Reading**\n","\n","> Bolouri, H.  Computational Modeling of Gene Regulatory Networks – A Primer.  Imperial College Press, 2008.\n","\n","> Gardner, T. S., Cantor, C. R., Collins, J. J.  Construction of a genetic toggle switch in Escherichia coli.  Nature.  2000.  403(6767): 339-42.\n","\n","> Selinger, D. W., Saxena, R. M., Cheung, K. J., Church G. M., Rosenow, C.  Global RNA half-life analysis in Escherichia coli reveals positional patterns of transcript degradation.  Genome Research.  2003.  13(2): 216-223.\n","\n","> Thomas, R.  Boolean formalization of genetic control circuits.  Journal of Theoretical Biology.  1973.  42(3): 563-85."],"metadata":{"id":"lH-znQk-5aqk"}},{"cell_type":"markdown","source":["## **Problems**"],"metadata":{"id":"0GWeL66N6sUD"}},{"cell_type":"markdown","source":["### ***Problem 2.1: Boolean analysis of a positive feedback network***"],"metadata":{"id":"yCpf0R81635Q"}},{"cell_type":"markdown","source":["![HW 2.1](https://drive.google.com/uc?export=view&id=1Ohe9Ipxc6xl1ZbMk49gCT9Zq0Pi5WhtE)"],"metadata":{"id":"v_ng3kiB7hOE"}},{"cell_type":"markdown","source":["First, let’s practice with a system (shown above) that is very similar to the system in Figure 2.1, except that this system includes positive feedback instead of negative feedback.  Just as before, transcription produces mRNA, translation yields protein, and an activator must be present for translation to occur.  The gene and the protein must both be present in order for transcription to occur.  The inputs to this system are the gene and the activator.  You are interested in the changes in the mRNA and protein concentrations over time.\n","\n","**(a)**\t  Write out Boolean equations for each of the reactions and mRNA and protein.\n","\n","**(b)**\t  Construct the state diagram of your system.  Circle all of the stable states.\n","\n","**(c)**\t  Beginning with initial conditions Protein = 1, mRNA = 0, Gene = 1, and Activator = 0, plot the dynamics of the system.  Label the changes that occur (for example, the onset of transcription).\n","\n","**(d)**\t  Imagine that the system starts with mRNA but no protein; both the gene and the activator are present.  Describe the progression of the system.  How does this progression change if you take the relative timing of protein and mRNA decay into account?\n"],"metadata":{"id":"kHgKB3-u7j_Z"}},{"cell_type":"markdown","source":["### ***Problem 2.2: Two interlocking regulatory circuits***"],"metadata":{"id":"PfZntRkN8HSS"}},{"cell_type":"markdown","source":["![HW 2.2](https://drive.google.com/uc?export=view&id=1imX_ORI1ulRcsSLqIuFwx3lySiJM8ADY)"],"metadata":{"id":"LqTms72O-nmd"}},{"cell_type":"markdown","source":["Now we will study a regulatory system where we have two proteins, each of which negatively regulates the expression of the other.  In this case, there are also two inhibitors, each of which blocks translation for one of the two mRNA types.  A diagram is shown above.\n","\n","Interestingly, this type of circuit, called the “toggle switch,” was the first **synthetic circuit** (see Gardner et al. in Recommended Reading).\n","\n","To pursue a Boolean analysis of this system, we will assume that the promoters and genes are always present and therefore need not be included in our equations or state matrix.  Inhibitor 1 and Inhibitor 2 will be our inputs.  There can be no translation of the first repressor if Inhibitor 1 is present, and no translation of Repressor 2 if Inhibitor 2 is present.  The molecules of interest will be mRNA 1, Repressor 1, mRNA 2, and\n","Repressor 2.  The processes we will track will be Transcription 1, Translation 1, Transcription 2, and Translation 2.\n","\n","**(a)**\tWrite out Boolean rules for all of the molecules of interest and processes for this system.\n","\n","**(b)**\tCalculate the state matrix.  Circle or highlight any stable states.\n","\n","**(c)**\tConsider the stable state in which Inhibitor 1 is present and Inhibitor 2 is not.  At time = 0, the culture medium is suddenly changed such that Inhibitor 2 is now present and Inhibitor 1 is not.  Draw a time plot, and describe the progression of the system in terms of state transitions.\n","\n","**(d)**\tUsing some of the typical time scales we discussed in this chapter, how long do you think the experiment described in (c) will take to reach a new stable state?\n"],"metadata":{"id":"Wjo-6cI09DoD"}},{"cell_type":"markdown","source":["### ***Problem 2.3: The lac operon regulatory network using Boolean logic***\n","\n","Now let’s apply our Boolean methods to the analysis of a real, and considerably more complicated, bacterial regulatory network.  The *lac* **operon** is required for the transport and metabolism of lactose in several enteric bacteria, including *E. coli*.  The *lac* operon is still used as a powerful model for investigating aspects of gene regulation and provides the basis for many synthetic gene circuits.  For more information on the pioneering work of Francois Jacob and Jacques Monod, who received the 1965 Nobel Prize in Medicine or Physiology for their work on gene regulation, see: http://symposium.cshlp.org/content/26/193.full.pdf+html\n","\n","The *lac* operon encodes three genes: *lacZ* (encoding the enzyme β-galactosidase, which cleaves lactose into glucose and galactose), *lacY* (the permease that transports lactose into the cell), and *lacA* (a transacetylase).  The operon is controlled primarily by the *lac* repressor (*LacI*), which prevents transcription in the absence of lactose, and the glucose-regulated activator CRP.  You will learn more about CRP in Chapters 7 and 10; for now, it’s sufficient to know that CRP induces expression of the *lac* genes and is inactivated by glucose.\n"],"metadata":{"id":"0swUzHAM9Sll"}},{"cell_type":"markdown","source":["![HW 2.3](https://drive.google.com/uc?export=view&id=1y6vbXqIe4mTT1wJ8X7TqoKJPxCQi_lRU)"],"metadata":{"id":"v8WT3C8F97J7"}},{"cell_type":"markdown","source":["In the simplified version of the *lac* network illustrated above, there is one gene ($G_{ZYA}$) that is transcribed and translated to produce protein $(P_{ZYA})$ in the presence of an activator ($CRP_{a}$; $CRP$ is active in the absence of glucose) and the absence of a negative regulator ($LacI_{a}$; $LacI$ is active in the absence of lactose).  $P_{ZYA}$ transports lactose into the cell if external lactose $(Lactose_{ext})$ is present.\n","\n","The two inputs into this system are glucose and external lactose.\n","\n","The four quantities that we wish to track are $CRP_{a}$, and $LacI_{a}$, internal lactose $(Lactose_{int})$, and $P_{ZYA}$.  The four processes that we care about are the transport of external lactose into the cell, the synthesis of $P_{ZYA}$ (we will lump transcription and translation into one process in this model), and the activation of $LacI$ ($LacI_{i}$ to $LacI_{a}$) and CRP ($CRP_{i}$ to $CRP_{a}$).\n","\n","Assume that the inactive regulators, $CRP_{i}$ and $LacI_{i}$, are present at stable levels inside the cell, and that $G_{ZYA}$ is always present.  These assumptions allow you to remove these quantities from your equations.\n","\n","**(a)**\tInitially, assume that activation of $CRP$ and LacI happens much faster than transcription or transport.  These assumptions allow you to eliminate two quantities and two processes from the system of equations.  Write your equations for this simplified model.\n","\n","**(b)**\tBuild and fill in your state diagram.  You may do this by hand or in Excel or MATLAB (Excel may be more useful in this particular case).  Highlight the stable states of the system.\n","\n","**(c)**\tIs the system capable of oscillating?  Identify any cyclic behavior and explain the logic behind it in biological terms.\n","\n","**(d)**\tAssume now that lactose transport across the membrane is slightly leaky.  What will happen to the oscillations from (c) when external lactose is present in the absence of glucose?  Could this leakiness be a useful attribute of the real system?  Explain why or why not in biological terms.\n","\n","**(e)**\tNow consider the network without assuming instantaneous activation of $CRP$ and $LacI_{i}$.  What is your new set of equations?\n","\n","**(f)**\tBuild and show your new complete state diagram for the system in (e).\n","\n","**(g)**\tHighlight the stable states of the system in (e).  Can this new system oscillate?  Explain why this network behaves differently from the system in (c) under conditions of abundant external lactose and no glucose.  Was the initial assumption to separate the timescales a reasonable one?  Justify your answer.\n"],"metadata":{"id":"NyVu3vqQ9k_s"}},{"cell_type":"markdown","source":["### ***Problem 2.4.  Bacterial chemotaxis***"],"metadata":{"id":"0bcsJgtK98xT"}},{"cell_type":"markdown","source":["![HW 2.4](https://drive.google.com/uc?export=view&id=1cv4NPPGnbISNmoO__eKL-FWAha2jhHVA)"],"metadata":{"id":"CsMLifHi-hUc"}},{"cell_type":"markdown","source":["So far, we have focused on the regulation of gene expression, but the Boolean approach can also be used to analyze the dynamics of other biological circuits such as signaling networks, protein kinase cascades, and metabolic pathways.  Here we will analyze a well-studied phenomenon: bacterial chemotaxis.  In the lab, bacteria are most often cultured in homogenous environments that provide all of the nutrients and energy sources that the microorganisms need to grow and proliferate.  However, in nature the environment in which the bacteria reside can be quite heterogeneous and dynamic.  Bacteria need to be able to quickly sense the conditions of the surrounding environment and move toward food sources and/or away from dangerous chemicals or toxins.  This process is called “chemotaxis.”  This video gives a straightforward overview of the process:\n","\n","http://www.youtube.com/watch?v=1wW2CZz6nM4&feature=related\n","\n","Another interesting video tracks the motion of *E. coli* during chemotaxis:\n","\n","http://www.youtube.com/watch?v=EZ5ATNJfuCs\n","\n","Here, we will investigate the dynamics of bacterial motion using Boolean modeling.  The above figure is a simplified representation of the chemotaxis network in *E. coli*.  The methyl-accepting chemotaxis protein (MCP) is a transmembrane protein that binds potential food sources in the environment (stimulus).  In the absence of these signals, the $CheA$ protein autophosphorylates itself and transfers this phosphate group to the $CheY$ and $CheR$ proteins.  The phophorylated “active” form of $CheY$ stimulates clockwise motion of the *E. coli* flagellum, which leads to tumbling.  In contrast, when $CheR$ is phophorylated, it is inactive.  When MCP is bound to a stimulus, it inhibits the kinase activity of $CheA$, leaving $CheY$ in the inactive form and $CheR$ in the active form; the bacterium therefore “runs.”  However, $CheR$ can also methylate the MCP receptor, which can cause $CheA$ to regain its phosphorylation activity.  This system therefore exhibits a form of negative feedback.\n","\n","Assume that all of the activation and repression steps happen after some time.  Also assume that $CheA$ and $CheR$ have a default active state (they are activated in the absence of the repressor) and that the repression of MCP by $CheR$ overcomes activation by a stimulus.\n","\n","We will model four processes of this system: activation of MCP ($ActMCP$), activation of $CheA$ $(ActCheA)$ activation of $CheY$ $(ActCheY)$, and activation of $CheR$ $(ActCheR)$.  You are interested in the active forms of MCP, $CheA$, $CheY$, and $CheR$.  We have one constant input into the system: the stimulus.\n","\n","**(a)**\tWrite out equations for each of the reactions and molecules of interest.\n","\n","**(b)**\tCalculate the state matrix, and highlight any stable states.\n","\n","**(c)**\tAssume that your system starts from a state where no stimulus is present and none of the proteins of interest are activated.  Trace out the dynamics of the system on your state matrix as the system moves from state to state.  What happens to the system?  What does your observation imply about bacterial chemotaxis in the absence of a stimulus?\n","\n","**(d)**\tStarting from the state that you ended with in (c), assume that a stimulus is suddenly added to the environment.  What happens to the network?  Trace the new dynamics on a state matrix, and graph the behavior we would expect the bacterium to exhibit (running or tumbling) over time.\n","\n","**(e)**\tNow assume that the bacterium has a mutation in MCP that prevents its negative regulation by $CheR$, such that:\n","\n","> **<h3>$ActMCP$ = IF Stimulus**\n","\n","Draw a new state matrix corresponding to this new set of rules, determine the stable state of the new system with no stimulus present, and then trace the dynamics of the system upon addition of stimulus using a state matrix.  How is your answer different from (d)?  What is the purpose of $CheR$’s negative regulation of MCP activity?\n"],"metadata":{"id":"6p759Vi5-A56"}}]}